{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "authorship_tag": "ABX9TyNzzDMbSe9dO52SfUkNkGT6",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Mai-00048/Ranked-vaccine-peptides/blob/main/Untitled44.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Oz2gVVUxJhL6",
        "outputId": "90a52ecf-30bb-45f1-e487-32631d135aaa"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'vaxrank'...\n",
            "remote: Enumerating objects: 2126, done.\u001b[K\n",
            "remote: Counting objects: 100% (244/244), done.\u001b[K\n",
            "remote: Compressing objects: 100% (142/142), done.\u001b[K\n",
            "remote: Total 2126 (delta 134), reused 177 (delta 77), pack-reused 1882 (from 1)\u001b[K\n",
            "Receiving objects: 100% (2126/2126), 1009.91 KiB | 12.02 MiB/s, done.\n",
            "Resolving deltas: 100% (1393/1393), done.\n",
            "/content/vaxrank\n"
          ]
        }
      ],
      "source": [
        "!git clone https://github.com/openvax/vaxrank.git\n",
        "%cd vaxrank"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install -r requirements.txt\n",
        "!pip install .  # تثبيت Vaxrank كحزمة"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Gw1Y5D7IJkut",
        "outputId": "a1793382-e25a-46a8-8c72-68d148d760f8"
      },
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: six in /usr/local/lib/python3.10/dist-packages (from -r requirements.txt (line 1)) (1.17.0)\n",
            "Requirement already satisfied: numpy>=1.14.0 in /usr/local/lib/python3.10/dist-packages (from -r requirements.txt (line 2)) (1.26.4)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.10/dist-packages (from -r requirements.txt (line 3)) (2.2.2)\n",
            "Collecting pyensembl>=1.5.0 (from -r requirements.txt (line 4))\n",
            "  Downloading pyensembl-2.3.13-py3-none-any.whl.metadata (9.4 kB)\n",
            "Collecting varcode>=0.5.9 (from -r requirements.txt (line 5))\n",
            "  Downloading varcode-1.2.1-py3-none-any.whl.metadata (9.2 kB)\n",
            "Collecting isovar>=1.1.1 (from -r requirements.txt (line 6))\n",
            "  Downloading isovar-1.3.0-py3-none-any.whl.metadata (18 kB)\n",
            "Collecting mhctools>=1.5.0 (from -r requirements.txt (line 7))\n",
            "  Downloading mhctools-1.9.0-py3-none-any.whl.metadata (4.6 kB)\n",
            "Collecting roman (from -r requirements.txt (line 8))\n",
            "  Downloading roman-4.2-py3-none-any.whl.metadata (3.6 kB)\n",
            "Collecting jinja2<3.1 (from -r requirements.txt (line 9))\n",
            "  Downloading Jinja2-3.0.3-py3-none-any.whl.metadata (3.5 kB)\n",
            "Collecting pdfkit (from -r requirements.txt (line 10))\n",
            "  Downloading pdfkit-1.0.0-py3-none-any.whl.metadata (9.3 kB)\n",
            "Collecting pypandoc (from -r requirements.txt (line 11))\n",
            "  Downloading pypandoc-1.14-py3-none-any.whl.metadata (16 kB)\n",
            "Collecting shellinford>=0.3.4 (from -r requirements.txt (line 12))\n",
            "  Downloading shellinford-0.4.1.tar.gz (64 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m65.0/65.0 kB\u001b[0m \u001b[31m4.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting xlsxwriter (from -r requirements.txt (line 13))\n",
            "  Downloading XlsxWriter-3.2.0-py3-none-any.whl.metadata (2.6 kB)\n",
            "Collecting xlrd<2.0.0,>=1.0.0 (from -r requirements.txt (line 14))\n",
            "  Downloading xlrd-1.2.0-py2.py3-none-any.whl.metadata (1.3 kB)\n",
            "Collecting xvfbwrapper (from -r requirements.txt (line 15))\n",
            "  Downloading xvfbwrapper-0.2.9.tar.gz (5.6 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: future>=0.16.0 in /usr/local/lib/python3.10/dist-packages (from -r requirements.txt (line 16)) (1.0.0)\n",
            "Requirement already satisfied: astropy in /usr/local/lib/python3.10/dist-packages (from -r requirements.txt (line 17)) (6.1.7)\n",
            "Collecting datacache (from -r requirements.txt (line 18))\n",
            "  Downloading datacache-1.4.1-py3-none-any.whl.metadata (1.9 kB)\n",
            "Collecting pysam>=0.15.2 (from -r requirements.txt (line 19))\n",
            "  Downloading pysam-0.22.1-cp310-cp310-manylinux_2_28_x86_64.whl.metadata (1.5 kB)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.10/dist-packages (from pandas->-r requirements.txt (line 3)) (2.8.2)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas->-r requirements.txt (line 3)) (2024.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.10/dist-packages (from pandas->-r requirements.txt (line 3)) (2024.2)\n",
            "Collecting typechecks<1.0.0,>=0.0.2 (from pyensembl>=1.5.0->-r requirements.txt (line 4))\n",
            "  Downloading typechecks-0.1.0.tar.gz (3.4 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting memoized-property>=1.0.2 (from pyensembl>=1.5.0->-r requirements.txt (line 4))\n",
            "  Downloading memoized-property-1.0.3.tar.gz (5.0 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting tinytimer<1.0.0,>=0.0.0 (from pyensembl>=1.5.0->-r requirements.txt (line 4))\n",
            "  Downloading tinytimer-0.0.0.tar.gz (2.1 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting gtfparse<3.0.0,>=2.5.0 (from pyensembl>=1.5.0->-r requirements.txt (line 4))\n",
            "  Downloading gtfparse-2.5.0-py3-none-any.whl.metadata (2.1 kB)\n",
            "Collecting serializable<1.0.0,>=0.2.1 (from pyensembl>=1.5.0->-r requirements.txt (line 4))\n",
            "  Downloading serializable-0.4.1-py3-none-any.whl.metadata (2.2 kB)\n",
            "Collecting pylint<3.0.0,>=2.17.2 (from pyensembl>=1.5.0->-r requirements.txt (line 4))\n",
            "  Downloading pylint-2.17.7-py3-none-any.whl.metadata (12 kB)\n",
            "Collecting biopython>=1.64 (from varcode>=0.5.9->-r requirements.txt (line 5))\n",
            "  Downloading biopython-1.84-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (12 kB)\n",
            "Collecting pyvcf3>=1.0.0 (from varcode>=0.5.9->-r requirements.txt (line 5))\n",
            "  Downloading PyVCF3-1.0.3.tar.gz (977 kB)\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m977.6/977.6 kB\u001b[0m \u001b[31m42.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting sercol>=0.1.4 (from varcode>=0.5.9->-r requirements.txt (line 5))\n",
            "  Downloading sercol-1.0.0-py3-none-any.whl.metadata (1.3 kB)\n",
            "Collecting cached-property>=1.5.1 (from isovar>=1.1.1->-r requirements.txt (line 6))\n",
            "  Downloading cached_property-2.0.1-py3-none-any.whl.metadata (10 kB)\n",
            "Requirement already satisfied: psutil in /usr/local/lib/python3.10/dist-packages (from isovar>=1.1.1->-r requirements.txt (line 6)) (5.9.5)\n",
            "Collecting mhcflurry>=2.0.0 (from mhctools>=1.5.0->-r requirements.txt (line 7))\n",
            "  Downloading mhcflurry-2.1.4-py3-none-any.whl.metadata (6.0 kB)\n",
            "Collecting mhcnames>=0.3.2 (from mhctools>=1.5.0->-r requirements.txt (line 7))\n",
            "  Downloading mhcnames-0.4.8.tar.gz (13 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.10/dist-packages (from jinja2<3.1->-r requirements.txt (line 9)) (3.0.2)\n",
            "Requirement already satisfied: pyerfa>=2.0.1.1 in /usr/local/lib/python3.10/dist-packages (from astropy->-r requirements.txt (line 17)) (2.0.1.5)\n",
            "Requirement already satisfied: astropy-iers-data>=0.2024.10.28.0.34.7 in /usr/local/lib/python3.10/dist-packages (from astropy->-r requirements.txt (line 17)) (0.2024.12.16.0.35.48)\n",
            "Requirement already satisfied: PyYAML>=3.13 in /usr/local/lib/python3.10/dist-packages (from astropy->-r requirements.txt (line 17)) (6.0.2)\n",
            "Requirement already satisfied: packaging>=19.0 in /usr/local/lib/python3.10/dist-packages (from astropy->-r requirements.txt (line 17)) (24.2)\n",
            "Collecting appdirs>=1.4.0 (from datacache->-r requirements.txt (line 18))\n",
            "  Downloading appdirs-1.4.4-py2.py3-none-any.whl.metadata (9.0 kB)\n",
            "Collecting progressbar33>=2.4 (from datacache->-r requirements.txt (line 18))\n",
            "  Downloading progressbar33-2.4.tar.gz (10 kB)\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: requests>=2.5.1 in /usr/local/lib/python3.10/dist-packages (from datacache->-r requirements.txt (line 18)) (2.32.3)\n",
            "Collecting mock (from datacache->-r requirements.txt (line 18))\n",
            "  Downloading mock-5.1.0-py3-none-any.whl.metadata (3.0 kB)\n",
            "Collecting polars<0.21.0,>=0.20.2 (from gtfparse<3.0.0,>=2.5.0->pyensembl>=1.5.0->-r requirements.txt (line 4))\n",
            "  Downloading polars-0.20.31-cp38-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (14 kB)\n",
            "Collecting pyarrow<14.1.0,>=14.0.2 (from gtfparse<3.0.0,>=2.5.0->pyensembl>=1.5.0->-r requirements.txt (line 4))\n",
            "  Downloading pyarrow-14.0.2-cp310-cp310-manylinux_2_28_x86_64.whl.metadata (3.0 kB)\n",
            "Requirement already satisfied: scikit-learn in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (1.6.0)\n",
            "Collecting mhcgnomes>=0.8.4 (from mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7))\n",
            "  Downloading mhcgnomes-1.8.6-py3-none-any.whl.metadata (8.2 kB)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (4.67.1)\n",
            "Requirement already satisfied: tensorflow>=2.15.0 in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (2.17.1)\n",
            "Requirement already satisfied: tf-keras in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (2.17.0)\n",
            "Requirement already satisfied: platformdirs>=2.2.0 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl>=1.5.0->-r requirements.txt (line 4)) (4.3.6)\n",
            "Collecting astroid<=2.17.0-dev0,>=2.15.8 (from pylint<3.0.0,>=2.17.2->pyensembl>=1.5.0->-r requirements.txt (line 4))\n",
            "  Downloading astroid-2.15.8-py3-none-any.whl.metadata (4.7 kB)\n",
            "Collecting isort<6,>=4.2.5 (from pylint<3.0.0,>=2.17.2->pyensembl>=1.5.0->-r requirements.txt (line 4))\n",
            "  Downloading isort-5.13.2-py3-none-any.whl.metadata (12 kB)\n",
            "Collecting mccabe<0.8,>=0.6 (from pylint<3.0.0,>=2.17.2->pyensembl>=1.5.0->-r requirements.txt (line 4))\n",
            "  Downloading mccabe-0.7.0-py2.py3-none-any.whl.metadata (5.0 kB)\n",
            "Collecting tomlkit>=0.10.1 (from pylint<3.0.0,>=2.17.2->pyensembl>=1.5.0->-r requirements.txt (line 4))\n",
            "  Downloading tomlkit-0.13.2-py3-none-any.whl.metadata (2.7 kB)\n",
            "Collecting dill>=0.2 (from pylint<3.0.0,>=2.17.2->pyensembl>=1.5.0->-r requirements.txt (line 4))\n",
            "  Downloading dill-0.3.9-py3-none-any.whl.metadata (10 kB)\n",
            "Requirement already satisfied: tomli>=1.1.0 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl>=1.5.0->-r requirements.txt (line 4)) (2.2.1)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.10/dist-packages (from pyvcf3>=1.0.0->varcode>=0.5.9->-r requirements.txt (line 5)) (75.1.0)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests>=2.5.1->datacache->-r requirements.txt (line 18)) (3.4.0)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests>=2.5.1->datacache->-r requirements.txt (line 18)) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests>=2.5.1->datacache->-r requirements.txt (line 18)) (2.2.3)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests>=2.5.1->datacache->-r requirements.txt (line 18)) (2024.12.14)\n",
            "Collecting simplejson (from sercol>=0.1.4->varcode>=0.5.9->-r requirements.txt (line 5))\n",
            "  Downloading simplejson-3.19.3-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.2 kB)\n",
            "Collecting lazy-object-proxy>=1.4.0 (from astroid<=2.17.0-dev0,>=2.15.8->pylint<3.0.0,>=2.17.2->pyensembl>=1.5.0->-r requirements.txt (line 4))\n",
            "  Downloading lazy_object_proxy-1.10.0-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.8 kB)\n",
            "Requirement already satisfied: wrapt<2,>=1.11 in /usr/local/lib/python3.10/dist-packages (from astroid<=2.17.0-dev0,>=2.15.8->pylint<3.0.0,>=2.17.2->pyensembl>=1.5.0->-r requirements.txt (line 4)) (1.17.0)\n",
            "Requirement already satisfied: typing-extensions>=4.0.0 in /usr/local/lib/python3.10/dist-packages (from astroid<=2.17.0-dev0,>=2.15.8->pylint<3.0.0,>=2.17.2->pyensembl>=1.5.0->-r requirements.txt (line 4)) (4.12.2)\n",
            "Requirement already satisfied: absl-py>=1.0.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (1.4.0)\n",
            "Requirement already satisfied: astunparse>=1.6.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (1.6.3)\n",
            "Requirement already satisfied: flatbuffers>=24.3.25 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (24.3.25)\n",
            "Requirement already satisfied: gast!=0.5.0,!=0.5.1,!=0.5.2,>=0.2.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (0.6.0)\n",
            "Requirement already satisfied: google-pasta>=0.1.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (0.2.0)\n",
            "Requirement already satisfied: h5py>=3.10.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (3.12.1)\n",
            "Requirement already satisfied: libclang>=13.0.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (18.1.1)\n",
            "Requirement already satisfied: ml-dtypes<0.5.0,>=0.3.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (0.4.1)\n",
            "Requirement already satisfied: opt-einsum>=2.3.2 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (3.4.0)\n",
            "Requirement already satisfied: protobuf!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0dev,>=3.20.3 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (4.25.5)\n",
            "Requirement already satisfied: termcolor>=1.1.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (2.5.0)\n",
            "Requirement already satisfied: grpcio<2.0,>=1.24.3 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (1.68.1)\n",
            "Requirement already satisfied: tensorboard<2.18,>=2.17 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (2.17.1)\n",
            "Requirement already satisfied: keras>=3.2.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (3.5.0)\n",
            "Requirement already satisfied: tensorflow-io-gcs-filesystem>=0.23.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (0.37.1)\n",
            "Requirement already satisfied: scipy>=1.6.0 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (1.13.1)\n",
            "Requirement already satisfied: joblib>=1.2.0 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (1.4.2)\n",
            "Requirement already satisfied: threadpoolctl>=3.1.0 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (3.5.0)\n",
            "Requirement already satisfied: wheel<1.0,>=0.23.0 in /usr/local/lib/python3.10/dist-packages (from astunparse>=1.6.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (0.45.1)\n",
            "Requirement already satisfied: rich in /usr/local/lib/python3.10/dist-packages (from keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (13.9.4)\n",
            "Requirement already satisfied: namex in /usr/local/lib/python3.10/dist-packages (from keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (0.0.8)\n",
            "Requirement already satisfied: optree in /usr/local/lib/python3.10/dist-packages (from keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (0.13.1)\n",
            "Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.10/dist-packages (from tensorboard<2.18,>=2.17->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (3.7)\n",
            "Requirement already satisfied: tensorboard-data-server<0.8.0,>=0.7.0 in /usr/local/lib/python3.10/dist-packages (from tensorboard<2.18,>=2.17->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (0.7.2)\n",
            "Requirement already satisfied: werkzeug>=1.0.1 in /usr/local/lib/python3.10/dist-packages (from tensorboard<2.18,>=2.17->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (3.1.3)\n",
            "Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.10/dist-packages (from rich->keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (3.0.0)\n",
            "Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /usr/local/lib/python3.10/dist-packages (from rich->keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (2.18.0)\n",
            "Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.10/dist-packages (from markdown-it-py>=2.2.0->rich->keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools>=1.5.0->-r requirements.txt (line 7)) (0.1.2)\n",
            "Downloading pyensembl-2.3.13-py3-none-any.whl (55 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m56.0/56.0 kB\u001b[0m \u001b[31m4.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading varcode-1.2.1-py3-none-any.whl (120 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m120.4/120.4 kB\u001b[0m \u001b[31m9.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading isovar-1.3.0-py3-none-any.whl (110 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m110.5/110.5 kB\u001b[0m \u001b[31m8.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading mhctools-1.9.0-py3-none-any.whl (54 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m54.9/54.9 kB\u001b[0m \u001b[31m4.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading roman-4.2-py3-none-any.whl (5.5 kB)\n",
            "Downloading Jinja2-3.0.3-py3-none-any.whl (133 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m133.6/133.6 kB\u001b[0m \u001b[31m11.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pdfkit-1.0.0-py3-none-any.whl (12 kB)\n",
            "Downloading pypandoc-1.14-py3-none-any.whl (21 kB)\n",
            "Downloading XlsxWriter-3.2.0-py3-none-any.whl (159 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m159.9/159.9 kB\u001b[0m \u001b[31m12.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading xlrd-1.2.0-py2.py3-none-any.whl (103 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m103.3/103.3 kB\u001b[0m \u001b[31m9.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading datacache-1.4.1-py3-none-any.whl (20 kB)\n",
            "Downloading pysam-0.22.1-cp310-cp310-manylinux_2_28_x86_64.whl (22.0 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m22.0/22.0 MB\u001b[0m \u001b[31m67.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)\n",
            "Downloading biopython-1.84-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.2 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m3.2/3.2 MB\u001b[0m \u001b[31m86.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading cached_property-2.0.1-py3-none-any.whl (7.4 kB)\n",
            "Downloading gtfparse-2.5.0-py3-none-any.whl (15 kB)\n",
            "Downloading mhcflurry-2.1.4-py3-none-any.whl (140 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m140.8/140.8 kB\u001b[0m \u001b[31m10.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pylint-2.17.7-py3-none-any.whl (537 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m537.2/537.2 kB\u001b[0m \u001b[31m33.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading sercol-1.0.0-py3-none-any.whl (9.0 kB)\n",
            "Downloading serializable-0.4.1-py3-none-any.whl (12 kB)\n",
            "Downloading mock-5.1.0-py3-none-any.whl (30 kB)\n",
            "Downloading astroid-2.15.8-py3-none-any.whl (278 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m278.3/278.3 kB\u001b[0m \u001b[31m19.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading dill-0.3.9-py3-none-any.whl (119 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m119.4/119.4 kB\u001b[0m \u001b[31m10.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading isort-5.13.2-py3-none-any.whl (92 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m92.3/92.3 kB\u001b[0m \u001b[31m7.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading mccabe-0.7.0-py2.py3-none-any.whl (7.3 kB)\n",
            "Downloading mhcgnomes-1.8.6-py3-none-any.whl (103 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m103.7/103.7 kB\u001b[0m \u001b[31m8.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading polars-0.20.31-cp38-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (28.8 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m28.8/28.8 MB\u001b[0m \u001b[31m35.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pyarrow-14.0.2-cp310-cp310-manylinux_2_28_x86_64.whl (38.0 MB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m38.0/38.0 MB\u001b[0m \u001b[31m16.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading tomlkit-0.13.2-py3-none-any.whl (37 kB)\n",
            "Downloading simplejson-3.19.3-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (137 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m137.9/137.9 kB\u001b[0m \u001b[31m11.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading lazy_object_proxy-1.10.0-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (68 kB)\n",
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m68.3/68.3 kB\u001b[0m \u001b[31m5.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hBuilding wheels for collected packages: shellinford, xvfbwrapper, memoized-property, mhcnames, progressbar33, pyvcf3, tinytimer, typechecks\n",
            "  Building wheel for shellinford (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for shellinford: filename=shellinford-0.4.1-cp310-cp310-linux_x86_64.whl size=568762 sha256=2185425c966c20b0c58b2bba8b990638291568973922ccc6745cb7facdc1e4a9\n",
            "  Stored in directory: /root/.cache/pip/wheels/8d/13/e3/df135f38552e4f34c08a54fcccfcfb4902de23fd516c7e7826\n",
            "  Building wheel for xvfbwrapper (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for xvfbwrapper: filename=xvfbwrapper-0.2.9-py3-none-any.whl size=5010 sha256=486bb2356ba8ab034d1668472ea710c27556b44c7726e52d9c66ac2ae4759e85\n",
            "  Stored in directory: /root/.cache/pip/wheels/cb/10/7d/2b7fdffccf837f7d5425931575fbee9caebe2c190931f9058b\n",
            "  Building wheel for memoized-property (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for memoized-property: filename=memoized_property-1.0.3-py2.py3-none-any.whl size=4186 sha256=aae4763b664be85e6664fb2a7c5d11baf0f3fb783d38ee6a021da4c2ebabf4ec\n",
            "  Stored in directory: /root/.cache/pip/wheels/f7/0f/93/d0497c1248dcc9113cfa09e6ddf52f5636b68bcd6751bad581\n",
            "  Building wheel for mhcnames (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for mhcnames: filename=mhcnames-0.4.8-py3-none-any.whl size=15878 sha256=120ef644d4f8b2c107b3f74f3738777c61e68dca8f2ca20b14925086741012e9\n",
            "  Stored in directory: /root/.cache/pip/wheels/08/f4/83/1fff6ecb5cc9c9827ea5371e8812f351ffbfa14ff5f4354758\n",
            "  Building wheel for progressbar33 (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for progressbar33: filename=progressbar33-2.4-py3-none-any.whl size=12142 sha256=8d80c99bc669171cab13cdf1af0774dcb6c9450af3fce09895c1bb3bb28915d8\n",
            "  Stored in directory: /root/.cache/pip/wheels/28/6d/b6/fd4b9b1489353528a2eea1200c98532fd60d535e22498d9e28\n",
            "  Building wheel for pyvcf3 (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pyvcf3: filename=PyVCF3-1.0.3-cp310-cp310-linux_x86_64.whl size=1246727 sha256=ebe0f0f4cb6925d36c9a3f93492835406a14cc16e08b9d14ae24ececcb9904d8\n",
            "  Stored in directory: /root/.cache/pip/wheels/12/2e/72/c03483bc49d13b24ec16d0878e17df5e9d2cd9a088935acdda\n",
            "  Building wheel for tinytimer (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for tinytimer: filename=tinytimer-0.0.0-py3-none-any.whl size=1875 sha256=f12d3d0f3c177eecc33223a79f09b339ee18e8e09fff3a982aea1a756e26b295\n",
            "  Stored in directory: /root/.cache/pip/wheels/35/7f/a3/036dd96fcd3da5d509047bcc764790cea8f6c73edadb1cbedc\n",
            "  Building wheel for typechecks (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for typechecks: filename=typechecks-0.1.0-py3-none-any.whl size=2752 sha256=627fb110d8d37ab8d6b3070ec4003dd755e679c5238df91867f31432a7e773d9\n",
            "  Stored in directory: /root/.cache/pip/wheels/da/a7/4d/6fb19c879271cf359a24d5bc4d074d53380cc2473f70068c4e\n",
            "Successfully built shellinford xvfbwrapper memoized-property mhcnames progressbar33 pyvcf3 tinytimer typechecks\n",
            "Installing collected packages: xvfbwrapper, typechecks, tinytimer, shellinford, progressbar33, pdfkit, memoized-property, appdirs, xlsxwriter, xlrd, tomlkit, simplejson, roman, pyvcf3, pysam, pypandoc, pyarrow, polars, mock, mhcnames, mccabe, lazy-object-proxy, jinja2, isort, dill, cached-property, biopython, serializable, gtfparse, astroid, sercol, pylint, mhcgnomes, datacache, pyensembl, varcode, mhcflurry, isovar, mhctools\n",
            "  Attempting uninstall: xlrd\n",
            "    Found existing installation: xlrd 2.0.1\n",
            "    Uninstalling xlrd-2.0.1:\n",
            "      Successfully uninstalled xlrd-2.0.1\n",
            "  Attempting uninstall: pyarrow\n",
            "    Found existing installation: pyarrow 17.0.0\n",
            "    Uninstalling pyarrow-17.0.0:\n",
            "      Successfully uninstalled pyarrow-17.0.0\n",
            "  Attempting uninstall: polars\n",
            "    Found existing installation: polars 1.9.0\n",
            "    Uninstalling polars-1.9.0:\n",
            "      Successfully uninstalled polars-1.9.0\n",
            "  Attempting uninstall: jinja2\n",
            "    Found existing installation: Jinja2 3.1.4\n",
            "    Uninstalling Jinja2-3.1.4:\n",
            "      Successfully uninstalled Jinja2-3.1.4\n",
            "\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n",
            "flask 3.1.0 requires Jinja2>=3.1.2, but you have jinja2 3.0.3 which is incompatible.\n",
            "sphinx 8.1.3 requires Jinja2>=3.1, but you have jinja2 3.0.3 which is incompatible.\u001b[0m\u001b[31m\n",
            "\u001b[0mSuccessfully installed appdirs-1.4.4 astroid-2.15.8 biopython-1.84 cached-property-2.0.1 datacache-1.4.1 dill-0.3.9 gtfparse-2.5.0 isort-5.13.2 isovar-1.3.0 jinja2-3.0.3 lazy-object-proxy-1.10.0 mccabe-0.7.0 memoized-property-1.0.3 mhcflurry-2.1.4 mhcgnomes-1.8.6 mhcnames-0.4.8 mhctools-1.9.0 mock-5.1.0 pdfkit-1.0.0 polars-0.20.31 progressbar33-2.4 pyarrow-14.0.2 pyensembl-2.3.13 pylint-2.17.7 pypandoc-1.14 pysam-0.22.1 pyvcf3-1.0.3 roman-4.2 sercol-1.0.0 serializable-0.4.1 shellinford-0.4.1 simplejson-3.19.3 tinytimer-0.0.0 tomlkit-0.13.2 typechecks-0.1.0 varcode-1.2.1 xlrd-1.2.0 xlsxwriter-3.2.0 xvfbwrapper-0.2.9\n",
            "Processing /content/vaxrank\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: numpy<2.0.0,>=1.14.0 in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (1.26.4)\n",
            "Requirement already satisfied: pandas<3.0.0,>=2.1.4 in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (2.2.2)\n",
            "Requirement already satisfied: pyensembl<3.0.0,>=2.0.0 in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (2.3.13)\n",
            "Requirement already satisfied: varcode<2.0.0,>=1.1.0 in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (1.2.1)\n",
            "Requirement already satisfied: isovar<2.0.0,>=1.3.0 in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (1.3.0)\n",
            "Requirement already satisfied: mhctools<2.0.0,>=1.8.2 in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (1.9.0)\n",
            "Requirement already satisfied: roman in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (4.2)\n",
            "Requirement already satisfied: jinja2<3.1 in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (3.0.3)\n",
            "Requirement already satisfied: pdfkit in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (1.0.0)\n",
            "Requirement already satisfied: pypandoc in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (1.14)\n",
            "Requirement already satisfied: shellinford>=0.3.4 in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (0.4.1)\n",
            "Requirement already satisfied: xlrd<2.0.0,>=1.0.0 in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (1.2.0)\n",
            "Requirement already satisfied: xlsxwriter in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (3.2.0)\n",
            "Requirement already satisfied: xvfbwrapper in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (0.2.9)\n",
            "Requirement already satisfied: future>=0.16.0 in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (1.0.0)\n",
            "Requirement already satisfied: astropy in /usr/local/lib/python3.10/dist-packages (from vaxrank==1.5.0) (6.1.7)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.10/dist-packages (from isovar<2.0.0,>=1.3.0->vaxrank==1.5.0) (1.17.0)\n",
            "Requirement already satisfied: pysam>=0.15.2 in /usr/local/lib/python3.10/dist-packages (from isovar<2.0.0,>=1.3.0->vaxrank==1.5.0) (0.22.1)\n",
            "Requirement already satisfied: cached-property>=1.5.1 in /usr/local/lib/python3.10/dist-packages (from isovar<2.0.0,>=1.3.0->vaxrank==1.5.0) (2.0.1)\n",
            "Requirement already satisfied: psutil in /usr/local/lib/python3.10/dist-packages (from isovar<2.0.0,>=1.3.0->vaxrank==1.5.0) (5.9.5)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.10/dist-packages (from jinja2<3.1->vaxrank==1.5.0) (3.0.2)\n",
            "Requirement already satisfied: sercol>=0.0.2 in /usr/local/lib/python3.10/dist-packages (from mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (1.0.0)\n",
            "Requirement already satisfied: mhcflurry>=2.0.0 in /usr/local/lib/python3.10/dist-packages (from mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (2.1.4)\n",
            "Requirement already satisfied: mhcnames>=0.3.2 in /usr/local/lib/python3.10/dist-packages (from mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (0.4.8)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.10/dist-packages (from pandas<3.0.0,>=2.1.4->vaxrank==1.5.0) (2.8.2)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas<3.0.0,>=2.1.4->vaxrank==1.5.0) (2024.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.10/dist-packages (from pandas<3.0.0,>=2.1.4->vaxrank==1.5.0) (2024.2)\n",
            "Requirement already satisfied: typechecks<1.0.0,>=0.0.2 in /usr/local/lib/python3.10/dist-packages (from pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (0.1.0)\n",
            "Requirement already satisfied: datacache<2.0.0,>=1.4.0 in /usr/local/lib/python3.10/dist-packages (from pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (1.4.1)\n",
            "Requirement already satisfied: memoized-property>=1.0.2 in /usr/local/lib/python3.10/dist-packages (from pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (1.0.3)\n",
            "Requirement already satisfied: tinytimer<1.0.0,>=0.0.0 in /usr/local/lib/python3.10/dist-packages (from pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (0.0.0)\n",
            "Requirement already satisfied: gtfparse<3.0.0,>=2.5.0 in /usr/local/lib/python3.10/dist-packages (from pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (2.5.0)\n",
            "Requirement already satisfied: serializable<1.0.0,>=0.2.1 in /usr/local/lib/python3.10/dist-packages (from pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (0.4.1)\n",
            "Requirement already satisfied: pylint<3.0.0,>=2.17.2 in /usr/local/lib/python3.10/dist-packages (from pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (2.17.7)\n",
            "Requirement already satisfied: biopython>=1.64 in /usr/local/lib/python3.10/dist-packages (from varcode<2.0.0,>=1.1.0->vaxrank==1.5.0) (1.84)\n",
            "Requirement already satisfied: pyvcf3>=1.0.0 in /usr/local/lib/python3.10/dist-packages (from varcode<2.0.0,>=1.1.0->vaxrank==1.5.0) (1.0.3)\n",
            "Requirement already satisfied: pyerfa>=2.0.1.1 in /usr/local/lib/python3.10/dist-packages (from astropy->vaxrank==1.5.0) (2.0.1.5)\n",
            "Requirement already satisfied: astropy-iers-data>=0.2024.10.28.0.34.7 in /usr/local/lib/python3.10/dist-packages (from astropy->vaxrank==1.5.0) (0.2024.12.16.0.35.48)\n",
            "Requirement already satisfied: PyYAML>=3.13 in /usr/local/lib/python3.10/dist-packages (from astropy->vaxrank==1.5.0) (6.0.2)\n",
            "Requirement already satisfied: packaging>=19.0 in /usr/local/lib/python3.10/dist-packages (from astropy->vaxrank==1.5.0) (24.2)\n",
            "Requirement already satisfied: appdirs>=1.4.0 in /usr/local/lib/python3.10/dist-packages (from datacache<2.0.0,>=1.4.0->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (1.4.4)\n",
            "Requirement already satisfied: progressbar33>=2.4 in /usr/local/lib/python3.10/dist-packages (from datacache<2.0.0,>=1.4.0->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (2.4)\n",
            "Requirement already satisfied: requests>=2.5.1 in /usr/local/lib/python3.10/dist-packages (from datacache<2.0.0,>=1.4.0->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (2.32.3)\n",
            "Requirement already satisfied: mock in /usr/local/lib/python3.10/dist-packages (from datacache<2.0.0,>=1.4.0->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (5.1.0)\n",
            "Requirement already satisfied: polars<0.21.0,>=0.20.2 in /usr/local/lib/python3.10/dist-packages (from gtfparse<3.0.0,>=2.5.0->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (0.20.31)\n",
            "Requirement already satisfied: pyarrow<14.1.0,>=14.0.2 in /usr/local/lib/python3.10/dist-packages (from gtfparse<3.0.0,>=2.5.0->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (14.0.2)\n",
            "Requirement already satisfied: scikit-learn in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (1.6.0)\n",
            "Requirement already satisfied: mhcgnomes>=0.8.4 in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (1.8.6)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (4.67.1)\n",
            "Requirement already satisfied: tensorflow>=2.15.0 in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (2.17.1)\n",
            "Requirement already satisfied: tf-keras in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (2.17.0)\n",
            "Requirement already satisfied: platformdirs>=2.2.0 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (4.3.6)\n",
            "Requirement already satisfied: astroid<=2.17.0-dev0,>=2.15.8 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (2.15.8)\n",
            "Requirement already satisfied: isort<6,>=4.2.5 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (5.13.2)\n",
            "Requirement already satisfied: mccabe<0.8,>=0.6 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (0.7.0)\n",
            "Requirement already satisfied: tomlkit>=0.10.1 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (0.13.2)\n",
            "Requirement already satisfied: dill>=0.2 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (0.3.9)\n",
            "Requirement already satisfied: tomli>=1.1.0 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (2.2.1)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.10/dist-packages (from pyvcf3>=1.0.0->varcode<2.0.0,>=1.1.0->vaxrank==1.5.0) (75.1.0)\n",
            "Requirement already satisfied: simplejson in /usr/local/lib/python3.10/dist-packages (from sercol>=0.0.2->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (3.19.3)\n",
            "Requirement already satisfied: lazy-object-proxy>=1.4.0 in /usr/local/lib/python3.10/dist-packages (from astroid<=2.17.0-dev0,>=2.15.8->pylint<3.0.0,>=2.17.2->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (1.10.0)\n",
            "Requirement already satisfied: wrapt<2,>=1.11 in /usr/local/lib/python3.10/dist-packages (from astroid<=2.17.0-dev0,>=2.15.8->pylint<3.0.0,>=2.17.2->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (1.17.0)\n",
            "Requirement already satisfied: typing-extensions>=4.0.0 in /usr/local/lib/python3.10/dist-packages (from astroid<=2.17.0-dev0,>=2.15.8->pylint<3.0.0,>=2.17.2->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (4.12.2)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests>=2.5.1->datacache<2.0.0,>=1.4.0->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (3.4.0)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests>=2.5.1->datacache<2.0.0,>=1.4.0->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests>=2.5.1->datacache<2.0.0,>=1.4.0->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (2.2.3)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests>=2.5.1->datacache<2.0.0,>=1.4.0->pyensembl<3.0.0,>=2.0.0->vaxrank==1.5.0) (2024.12.14)\n",
            "Requirement already satisfied: absl-py>=1.0.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (1.4.0)\n",
            "Requirement already satisfied: astunparse>=1.6.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (1.6.3)\n",
            "Requirement already satisfied: flatbuffers>=24.3.25 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (24.3.25)\n",
            "Requirement already satisfied: gast!=0.5.0,!=0.5.1,!=0.5.2,>=0.2.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (0.6.0)\n",
            "Requirement already satisfied: google-pasta>=0.1.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (0.2.0)\n",
            "Requirement already satisfied: h5py>=3.10.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (3.12.1)\n",
            "Requirement already satisfied: libclang>=13.0.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (18.1.1)\n",
            "Requirement already satisfied: ml-dtypes<0.5.0,>=0.3.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (0.4.1)\n",
            "Requirement already satisfied: opt-einsum>=2.3.2 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (3.4.0)\n",
            "Requirement already satisfied: protobuf!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0dev,>=3.20.3 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (4.25.5)\n",
            "Requirement already satisfied: termcolor>=1.1.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (2.5.0)\n",
            "Requirement already satisfied: grpcio<2.0,>=1.24.3 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (1.68.1)\n",
            "Requirement already satisfied: tensorboard<2.18,>=2.17 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (2.17.1)\n",
            "Requirement already satisfied: keras>=3.2.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (3.5.0)\n",
            "Requirement already satisfied: tensorflow-io-gcs-filesystem>=0.23.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (0.37.1)\n",
            "Requirement already satisfied: scipy>=1.6.0 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (1.13.1)\n",
            "Requirement already satisfied: joblib>=1.2.0 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (1.4.2)\n",
            "Requirement already satisfied: threadpoolctl>=3.1.0 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (3.5.0)\n",
            "Requirement already satisfied: wheel<1.0,>=0.23.0 in /usr/local/lib/python3.10/dist-packages (from astunparse>=1.6.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (0.45.1)\n",
            "Requirement already satisfied: rich in /usr/local/lib/python3.10/dist-packages (from keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (13.9.4)\n",
            "Requirement already satisfied: namex in /usr/local/lib/python3.10/dist-packages (from keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (0.0.8)\n",
            "Requirement already satisfied: optree in /usr/local/lib/python3.10/dist-packages (from keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (0.13.1)\n",
            "Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.10/dist-packages (from tensorboard<2.18,>=2.17->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (3.7)\n",
            "Requirement already satisfied: tensorboard-data-server<0.8.0,>=0.7.0 in /usr/local/lib/python3.10/dist-packages (from tensorboard<2.18,>=2.17->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (0.7.2)\n",
            "Requirement already satisfied: werkzeug>=1.0.1 in /usr/local/lib/python3.10/dist-packages (from tensorboard<2.18,>=2.17->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (3.1.3)\n",
            "Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.10/dist-packages (from rich->keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (3.0.0)\n",
            "Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /usr/local/lib/python3.10/dist-packages (from rich->keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (2.18.0)\n",
            "Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.10/dist-packages (from markdown-it-py>=2.2.0->rich->keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools<2.0.0,>=1.8.2->vaxrank==1.5.0) (0.1.2)\n",
            "Building wheels for collected packages: vaxrank\n",
            "  Building wheel for vaxrank (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for vaxrank: filename=vaxrank-1.5.0-py3-none-any.whl size=55987 sha256=68d128ea56cb78b8df71b90a432cb036183bf0933def7875191d719c4ebbb9b9\n",
            "  Stored in directory: /tmp/pip-ephem-wheel-cache-xzn2sk6g/wheels/0c/ba/9e/658aad162afd69b585bd22b4cfaf8300b5081305669c020920\n",
            "Successfully built vaxrank\n",
            "Installing collected packages: vaxrank\n",
            "Successfully installed vaxrank-1.5.0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!sudo apt-get install -y wkhtmltopdf"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6h7BOmZfJnAJ",
        "outputId": "7feeb5ca-a3ba-4a9e-9282-8defaa3df33e"
      },
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  avahi-daemon bind9-host bind9-libs geoclue-2.0 glib-networking\n",
            "  glib-networking-common glib-networking-services gsettings-desktop-schemas\n",
            "  iio-sensor-proxy libavahi-core7 libavahi-glib1 libdaemon0 libevdev2\n",
            "  libfontenc1 libgudev-1.0-0 libhyphen0 libinput-bin libinput10\n",
            "  libjson-glib-1.0-0 libjson-glib-1.0-common liblmdb0 libmaxminddb0\n",
            "  libmbim-glib4 libmbim-proxy libmd4c0 libmm-glib0 libmtdev1 libnl-genl-3-200\n",
            "  libnotify4 libnss-mdns libproxy1v5 libqmi-glib5 libqmi-proxy libqt5core5a\n",
            "  libqt5dbus5 libqt5gui5 libqt5network5 libqt5positioning5 libqt5printsupport5\n",
            "  libqt5qml5 libqt5qmlmodels5 libqt5quick5 libqt5sensors5 libqt5svg5\n",
            "  libqt5webchannel5 libqt5webkit5 libqt5widgets5 libsoup2.4-1\n",
            "  libsoup2.4-common libudev1 libwacom-bin libwacom-common libwacom9 libwoff1\n",
            "  libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-render-util0 libxcb-util1\n",
            "  libxcb-xinerama0 libxcb-xinput0 libxcb-xkb1 libxfont2 libxkbcommon-x11-0\n",
            "  libxkbfile1 modemmanager qt5-gtk-platformtheme qttranslations5-l10n\n",
            "  session-migration systemd-hwe-hwdb udev usb-modeswitch usb-modeswitch-data\n",
            "  wpasupplicant x11-xkb-utils xfonts-base xfonts-encodings xfonts-utils xnest\n",
            "  xserver-common\n",
            "Suggested packages:\n",
            "  avahi-autoipd mmdb-bin gnome-shell | notification-daemon avahi-autoipd\n",
            "  | zeroconf qt5-image-formats-plugins qtwayland5 qt5-qmltooling-plugins comgt\n",
            "  wvdial wpagui libengine-pkcs11-openssl\n",
            "The following NEW packages will be installed:\n",
            "  avahi-daemon bind9-host bind9-libs geoclue-2.0 glib-networking\n",
            "  glib-networking-common glib-networking-services gsettings-desktop-schemas\n",
            "  iio-sensor-proxy libavahi-core7 libavahi-glib1 libdaemon0 libevdev2\n",
            "  libfontenc1 libgudev-1.0-0 libhyphen0 libinput-bin libinput10\n",
            "  libjson-glib-1.0-0 libjson-glib-1.0-common liblmdb0 libmaxminddb0\n",
            "  libmbim-glib4 libmbim-proxy libmd4c0 libmm-glib0 libmtdev1 libnl-genl-3-200\n",
            "  libnotify4 libnss-mdns libproxy1v5 libqmi-glib5 libqmi-proxy libqt5core5a\n",
            "  libqt5dbus5 libqt5gui5 libqt5network5 libqt5positioning5 libqt5printsupport5\n",
            "  libqt5qml5 libqt5qmlmodels5 libqt5quick5 libqt5sensors5 libqt5svg5\n",
            "  libqt5webchannel5 libqt5webkit5 libqt5widgets5 libsoup2.4-1\n",
            "  libsoup2.4-common libwacom-bin libwacom-common libwacom9 libwoff1\n",
            "  libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-render-util0 libxcb-util1\n",
            "  libxcb-xinerama0 libxcb-xinput0 libxcb-xkb1 libxfont2 libxkbcommon-x11-0\n",
            "  libxkbfile1 modemmanager qt5-gtk-platformtheme qttranslations5-l10n\n",
            "  session-migration systemd-hwe-hwdb udev usb-modeswitch usb-modeswitch-data\n",
            "  wkhtmltopdf wpasupplicant x11-xkb-utils xfonts-base xfonts-encodings\n",
            "  xfonts-utils xnest xserver-common\n",
            "The following packages will be upgraded:\n",
            "  libudev1\n",
            "1 upgraded, 80 newly installed, 0 to remove and 48 not upgraded.\n",
            "Need to get 44.6 MB of archives.\n",
            "After this operation, 156 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libavahi-core7 amd64 0.8-5ubuntu5.2 [90.8 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/main amd64 libdaemon0 amd64 0.14-7.1ubuntu3 [14.1 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/main amd64 liblmdb0 amd64 0.9.24-1build2 [47.6 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/main amd64 libmaxminddb0 amd64 1.5.2-1build2 [24.7 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 bind9-libs amd64 1:9.18.28-0ubuntu0.22.04.1 [1,256 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 bind9-host amd64 1:9.18.28-0ubuntu0.22.04.1 [52.6 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 avahi-daemon amd64 0.8-5ubuntu5.2 [69.7 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 libqt5core5a amd64 5.15.3+dfsg-2ubuntu0.2 [2,006 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu jammy/main amd64 libevdev2 amd64 1.12.1+dfsg-1 [39.5 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu jammy/main amd64 libmtdev1 amd64 1.1.6-1build4 [14.5 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libudev1 amd64 249.11-0ubuntu3.12 [78.2 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu jammy/main amd64 libgudev-1.0-0 amd64 1:237-2build1 [16.3 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu jammy/main amd64 libwacom-common all 2.2.0-1 [54.3 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu jammy/main amd64 libwacom9 amd64 2.2.0-1 [22.0 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libinput-bin amd64 1.20.0-1ubuntu0.3 [19.9 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libinput10 amd64 1.20.0-1ubuntu0.3 [131 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libmd4c0 amd64 0.4.8-1 [42.0 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 libqt5dbus5 amd64 5.15.3+dfsg-2ubuntu0.2 [222 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 libqt5network5 amd64 5.15.3+dfsg-2ubuntu0.2 [731 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxcb-icccm4 amd64 0.4.1-1.1build2 [11.5 kB]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxcb-util1 amd64 0.4.0-1build2 [11.4 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxcb-image0 amd64 0.4.0-2 [11.5 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxcb-keysyms1 amd64 0.4.0-1build3 [8,746 B]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxcb-render-util0 amd64 0.3.9-1build3 [10.3 kB]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxcb-xinerama0 amd64 1.14-3ubuntu3 [5,414 B]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxcb-xinput0 amd64 1.14-3ubuntu3 [34.3 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxcb-xkb1 amd64 1.14-3ubuntu3 [32.8 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxkbcommon-x11-0 amd64 1.4.0-1 [14.4 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 libqt5gui5 amd64 5.15.3+dfsg-2ubuntu0.2 [3,722 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 libqt5widgets5 amd64 5.15.3+dfsg-2ubuntu0.2 [2,561 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libqt5svg5 amd64 5.15.3-1 [149 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu jammy/main amd64 libhyphen0 amd64 2.8.8-7build2 [28.2 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libqt5positioning5 amd64 5.15.3+dfsg-3 [223 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 libqt5printsupport5 amd64 5.15.3+dfsg-2ubuntu0.2 [214 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libqt5qml5 amd64 5.15.3+dfsg-1 [1,472 kB]\n",
            "Get:36 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libqt5qmlmodels5 amd64 5.15.3+dfsg-1 [205 kB]\n",
            "Get:37 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libqt5quick5 amd64 5.15.3+dfsg-1 [1,748 kB]\n",
            "Get:38 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libqt5sensors5 amd64 5.15.3-1 [123 kB]\n",
            "Get:39 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libqt5webchannel5 amd64 5.15.3-1 [62.9 kB]\n",
            "Get:40 http://archive.ubuntu.com/ubuntu jammy/main amd64 libwoff1 amd64 1.0.2-1build4 [45.2 kB]\n",
            "Get:41 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libqt5webkit5 amd64 5.212.0~alpha4-15ubuntu1 [12.8 MB]\n",
            "Get:42 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 udev amd64 249.11-0ubuntu3.12 [1,557 kB]\n",
            "Get:43 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libavahi-glib1 amd64 0.8-5ubuntu5.2 [8,296 B]\n",
            "Get:44 http://archive.ubuntu.com/ubuntu jammy/main amd64 libjson-glib-1.0-common all 1.6.6-1build1 [4,432 B]\n",
            "Get:45 http://archive.ubuntu.com/ubuntu jammy/main amd64 libjson-glib-1.0-0 amd64 1.6.6-1build1 [69.9 kB]\n",
            "Get:46 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libmm-glib0 amd64 1.20.0-1~ubuntu22.04.4 [262 kB]\n",
            "Get:47 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libnotify4 amd64 0.7.9-3ubuntu5.22.04.1 [20.3 kB]\n",
            "Get:48 http://archive.ubuntu.com/ubuntu jammy/main amd64 libproxy1v5 amd64 0.4.17-2 [51.9 kB]\n",
            "Get:49 http://archive.ubuntu.com/ubuntu jammy/main amd64 glib-networking-common all 2.72.0-1 [3,718 B]\n",
            "Get:50 http://archive.ubuntu.com/ubuntu jammy/main amd64 glib-networking-services amd64 2.72.0-1 [9,982 B]\n",
            "Get:51 http://archive.ubuntu.com/ubuntu jammy/main amd64 session-migration amd64 0.3.6 [9,774 B]\n",
            "Get:52 http://archive.ubuntu.com/ubuntu jammy/main amd64 gsettings-desktop-schemas all 42.0-1ubuntu1 [31.1 kB]\n",
            "Get:53 http://archive.ubuntu.com/ubuntu jammy/main amd64 glib-networking amd64 2.72.0-1 [69.8 kB]\n",
            "Get:54 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libsoup2.4-common all 2.74.2-3ubuntu0.1 [4,070 B]\n",
            "Get:55 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libsoup2.4-1 amd64 2.74.2-3ubuntu0.1 [288 kB]\n",
            "Get:56 http://archive.ubuntu.com/ubuntu jammy/main amd64 geoclue-2.0 amd64 2.5.7-3ubuntu3 [111 kB]\n",
            "Get:57 http://archive.ubuntu.com/ubuntu jammy/main amd64 iio-sensor-proxy amd64 3.3-0ubuntu6 [34.4 kB]\n",
            "Get:58 http://archive.ubuntu.com/ubuntu jammy/main amd64 libfontenc1 amd64 1:1.1.4-1build3 [14.7 kB]\n",
            "Get:59 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libmbim-glib4 amd64 1.28.0-1~ubuntu20.04.1 [191 kB]\n",
            "Get:60 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libmbim-proxy amd64 1.28.0-1~ubuntu20.04.1 [6,130 B]\n",
            "Get:61 http://archive.ubuntu.com/ubuntu jammy/main amd64 libnl-genl-3-200 amd64 3.5.0-0.1 [12.4 kB]\n",
            "Get:62 http://archive.ubuntu.com/ubuntu jammy/main amd64 libnss-mdns amd64 0.15.1-1ubuntu1 [27.0 kB]\n",
            "Get:63 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libqmi-glib5 amd64 1.32.0-1ubuntu0.22.04.1 [772 kB]\n",
            "Get:64 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libqmi-proxy amd64 1.32.0-1ubuntu0.22.04.1 [6,072 B]\n",
            "Get:65 http://archive.ubuntu.com/ubuntu jammy/main amd64 libwacom-bin amd64 2.2.0-1 [13.6 kB]\n",
            "Get:66 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxfont2 amd64 1:2.0.5-1build1 [94.5 kB]\n",
            "Get:67 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxkbfile1 amd64 1:1.1.0-1build3 [71.8 kB]\n",
            "Get:68 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 modemmanager amd64 1.20.0-1~ubuntu22.04.4 [1,094 kB]\n",
            "Get:69 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 qt5-gtk-platformtheme amd64 5.15.3+dfsg-2ubuntu0.2 [130 kB]\n",
            "Get:70 http://archive.ubuntu.com/ubuntu jammy/universe amd64 qttranslations5-l10n all 5.15.3-1 [1,983 kB]\n",
            "Get:71 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 systemd-hwe-hwdb all 249.11.5 [3,228 B]\n",
            "Get:72 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 wpasupplicant amd64 2:2.10-6ubuntu2.1 [1,482 kB]\n",
            "Get:73 http://archive.ubuntu.com/ubuntu jammy/main amd64 x11-xkb-utils amd64 7.7+5build4 [172 kB]\n",
            "Get:74 http://archive.ubuntu.com/ubuntu jammy/main amd64 xfonts-encodings all 1:1.0.5-0ubuntu2 [578 kB]\n",
            "Get:75 http://archive.ubuntu.com/ubuntu jammy/main amd64 xfonts-utils amd64 1:7.7+6build2 [94.6 kB]\n",
            "Get:76 http://archive.ubuntu.com/ubuntu jammy/main amd64 xfonts-base all 1:1.0.5 [5,896 kB]\n",
            "Get:77 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 xserver-common all 2:21.1.4-2ubuntu1.7~22.04.12 [28.7 kB]\n",
            "Get:78 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 xnest amd64 2:21.1.4-2ubuntu1.7~22.04.12 [712 kB]\n",
            "Get:79 http://archive.ubuntu.com/ubuntu jammy/main amd64 usb-modeswitch-data all 20191128-4 [33.2 kB]\n",
            "Get:80 http://archive.ubuntu.com/ubuntu jammy/main amd64 usb-modeswitch amd64 2.6.1-3ubuntu2 [46.0 kB]\n",
            "Get:81 http://archive.ubuntu.com/ubuntu jammy/universe amd64 wkhtmltopdf amd64 0.12.6-2 [173 kB]\n",
            "Fetched 44.6 MB in 2s (19.6 MB/s)\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 78, <> line 81.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "dpkg-preconfigure: unable to re-open stdin: \n",
            "Selecting previously unselected package libavahi-core7:amd64.\n",
            "(Reading database ... 123634 files and directories currently installed.)\n",
            "Preparing to unpack .../00-libavahi-core7_0.8-5ubuntu5.2_amd64.deb ...\n",
            "Unpacking libavahi-core7:amd64 (0.8-5ubuntu5.2) ...\n",
            "Selecting previously unselected package libdaemon0:amd64.\n",
            "Preparing to unpack .../01-libdaemon0_0.14-7.1ubuntu3_amd64.deb ...\n",
            "Unpacking libdaemon0:amd64 (0.14-7.1ubuntu3) ...\n",
            "Selecting previously unselected package liblmdb0:amd64.\n",
            "Preparing to unpack .../02-liblmdb0_0.9.24-1build2_amd64.deb ...\n",
            "Unpacking liblmdb0:amd64 (0.9.24-1build2) ...\n",
            "Selecting previously unselected package libmaxminddb0:amd64.\n",
            "Preparing to unpack .../03-libmaxminddb0_1.5.2-1build2_amd64.deb ...\n",
            "Unpacking libmaxminddb0:amd64 (1.5.2-1build2) ...\n",
            "Selecting previously unselected package bind9-libs:amd64.\n",
            "Preparing to unpack .../04-bind9-libs_1%3a9.18.28-0ubuntu0.22.04.1_amd64.deb ...\n",
            "Unpacking bind9-libs:amd64 (1:9.18.28-0ubuntu0.22.04.1) ...\n",
            "Selecting previously unselected package bind9-host.\n",
            "Preparing to unpack .../05-bind9-host_1%3a9.18.28-0ubuntu0.22.04.1_amd64.deb ...\n",
            "Unpacking bind9-host (1:9.18.28-0ubuntu0.22.04.1) ...\n",
            "Selecting previously unselected package avahi-daemon.\n",
            "Preparing to unpack .../06-avahi-daemon_0.8-5ubuntu5.2_amd64.deb ...\n",
            "Unpacking avahi-daemon (0.8-5ubuntu5.2) ...\n",
            "Selecting previously unselected package libqt5core5a:amd64.\n",
            "Preparing to unpack .../07-libqt5core5a_5.15.3+dfsg-2ubuntu0.2_amd64.deb ...\n",
            "Unpacking libqt5core5a:amd64 (5.15.3+dfsg-2ubuntu0.2) ...\n",
            "Selecting previously unselected package libevdev2:amd64.\n",
            "Preparing to unpack .../08-libevdev2_1.12.1+dfsg-1_amd64.deb ...\n",
            "Unpacking libevdev2:amd64 (1.12.1+dfsg-1) ...\n",
            "Selecting previously unselected package libmtdev1:amd64.\n",
            "Preparing to unpack .../09-libmtdev1_1.1.6-1build4_amd64.deb ...\n",
            "Unpacking libmtdev1:amd64 (1.1.6-1build4) ...\n",
            "Preparing to unpack .../10-libudev1_249.11-0ubuntu3.12_amd64.deb ...\n",
            "Unpacking libudev1:amd64 (249.11-0ubuntu3.12) over (249.11-0ubuntu3.10) ...\n",
            "Setting up libudev1:amd64 (249.11-0ubuntu3.12) ...\n",
            "Selecting previously unselected package libgudev-1.0-0:amd64.\n",
            "(Reading database ... 123733 files and directories currently installed.)\n",
            "Preparing to unpack .../00-libgudev-1.0-0_1%3a237-2build1_amd64.deb ...\n",
            "Unpacking libgudev-1.0-0:amd64 (1:237-2build1) ...\n",
            "Selecting previously unselected package libwacom-common.\n",
            "Preparing to unpack .../01-libwacom-common_2.2.0-1_all.deb ...\n",
            "Unpacking libwacom-common (2.2.0-1) ...\n",
            "Selecting previously unselected package libwacom9:amd64.\n",
            "Preparing to unpack .../02-libwacom9_2.2.0-1_amd64.deb ...\n",
            "Unpacking libwacom9:amd64 (2.2.0-1) ...\n",
            "Selecting previously unselected package libinput-bin.\n",
            "Preparing to unpack .../03-libinput-bin_1.20.0-1ubuntu0.3_amd64.deb ...\n",
            "Unpacking libinput-bin (1.20.0-1ubuntu0.3) ...\n",
            "Selecting previously unselected package libinput10:amd64.\n",
            "Preparing to unpack .../04-libinput10_1.20.0-1ubuntu0.3_amd64.deb ...\n",
            "Unpacking libinput10:amd64 (1.20.0-1ubuntu0.3) ...\n",
            "Selecting previously unselected package libmd4c0:amd64.\n",
            "Preparing to unpack .../05-libmd4c0_0.4.8-1_amd64.deb ...\n",
            "Unpacking libmd4c0:amd64 (0.4.8-1) ...\n",
            "Selecting previously unselected package libqt5dbus5:amd64.\n",
            "Preparing to unpack .../06-libqt5dbus5_5.15.3+dfsg-2ubuntu0.2_amd64.deb ...\n",
            "Unpacking libqt5dbus5:amd64 (5.15.3+dfsg-2ubuntu0.2) ...\n",
            "Selecting previously unselected package libqt5network5:amd64.\n",
            "Preparing to unpack .../07-libqt5network5_5.15.3+dfsg-2ubuntu0.2_amd64.deb ...\n",
            "Unpacking libqt5network5:amd64 (5.15.3+dfsg-2ubuntu0.2) ...\n",
            "Selecting previously unselected package libxcb-icccm4:amd64.\n",
            "Preparing to unpack .../08-libxcb-icccm4_0.4.1-1.1build2_amd64.deb ...\n",
            "Unpacking libxcb-icccm4:amd64 (0.4.1-1.1build2) ...\n",
            "Selecting previously unselected package libxcb-util1:amd64.\n",
            "Preparing to unpack .../09-libxcb-util1_0.4.0-1build2_amd64.deb ...\n",
            "Unpacking libxcb-util1:amd64 (0.4.0-1build2) ...\n",
            "Selecting previously unselected package libxcb-image0:amd64.\n",
            "Preparing to unpack .../10-libxcb-image0_0.4.0-2_amd64.deb ...\n",
            "Unpacking libxcb-image0:amd64 (0.4.0-2) ...\n",
            "Selecting previously unselected package libxcb-keysyms1:amd64.\n",
            "Preparing to unpack .../11-libxcb-keysyms1_0.4.0-1build3_amd64.deb ...\n",
            "Unpacking libxcb-keysyms1:amd64 (0.4.0-1build3) ...\n",
            "Selecting previously unselected package libxcb-render-util0:amd64.\n",
            "Preparing to unpack .../12-libxcb-render-util0_0.3.9-1build3_amd64.deb ...\n",
            "Unpacking libxcb-render-util0:amd64 (0.3.9-1build3) ...\n",
            "Selecting previously unselected package libxcb-xinerama0:amd64.\n",
            "Preparing to unpack .../13-libxcb-xinerama0_1.14-3ubuntu3_amd64.deb ...\n",
            "Unpacking libxcb-xinerama0:amd64 (1.14-3ubuntu3) ...\n",
            "Selecting previously unselected package libxcb-xinput0:amd64.\n",
            "Preparing to unpack .../14-libxcb-xinput0_1.14-3ubuntu3_amd64.deb ...\n",
            "Unpacking libxcb-xinput0:amd64 (1.14-3ubuntu3) ...\n",
            "Selecting previously unselected package libxcb-xkb1:amd64.\n",
            "Preparing to unpack .../15-libxcb-xkb1_1.14-3ubuntu3_amd64.deb ...\n",
            "Unpacking libxcb-xkb1:amd64 (1.14-3ubuntu3) ...\n",
            "Selecting previously unselected package libxkbcommon-x11-0:amd64.\n",
            "Preparing to unpack .../16-libxkbcommon-x11-0_1.4.0-1_amd64.deb ...\n",
            "Unpacking libxkbcommon-x11-0:amd64 (1.4.0-1) ...\n",
            "Selecting previously unselected package libqt5gui5:amd64.\n",
            "Preparing to unpack .../17-libqt5gui5_5.15.3+dfsg-2ubuntu0.2_amd64.deb ...\n",
            "Unpacking libqt5gui5:amd64 (5.15.3+dfsg-2ubuntu0.2) ...\n",
            "Selecting previously unselected package libqt5widgets5:amd64.\n",
            "Preparing to unpack .../18-libqt5widgets5_5.15.3+dfsg-2ubuntu0.2_amd64.deb ...\n",
            "Unpacking libqt5widgets5:amd64 (5.15.3+dfsg-2ubuntu0.2) ...\n",
            "Selecting previously unselected package libqt5svg5:amd64.\n",
            "Preparing to unpack .../19-libqt5svg5_5.15.3-1_amd64.deb ...\n",
            "Unpacking libqt5svg5:amd64 (5.15.3-1) ...\n",
            "Selecting previously unselected package libhyphen0:amd64.\n",
            "Preparing to unpack .../20-libhyphen0_2.8.8-7build2_amd64.deb ...\n",
            "Unpacking libhyphen0:amd64 (2.8.8-7build2) ...\n",
            "Selecting previously unselected package libqt5positioning5:amd64.\n",
            "Preparing to unpack .../21-libqt5positioning5_5.15.3+dfsg-3_amd64.deb ...\n",
            "Unpacking libqt5positioning5:amd64 (5.15.3+dfsg-3) ...\n",
            "Selecting previously unselected package libqt5printsupport5:amd64.\n",
            "Preparing to unpack .../22-libqt5printsupport5_5.15.3+dfsg-2ubuntu0.2_amd64.deb ...\n",
            "Unpacking libqt5printsupport5:amd64 (5.15.3+dfsg-2ubuntu0.2) ...\n",
            "Selecting previously unselected package libqt5qml5:amd64.\n",
            "Preparing to unpack .../23-libqt5qml5_5.15.3+dfsg-1_amd64.deb ...\n",
            "Unpacking libqt5qml5:amd64 (5.15.3+dfsg-1) ...\n",
            "Selecting previously unselected package libqt5qmlmodels5:amd64.\n",
            "Preparing to unpack .../24-libqt5qmlmodels5_5.15.3+dfsg-1_amd64.deb ...\n",
            "Unpacking libqt5qmlmodels5:amd64 (5.15.3+dfsg-1) ...\n",
            "Selecting previously unselected package libqt5quick5:amd64.\n",
            "Preparing to unpack .../25-libqt5quick5_5.15.3+dfsg-1_amd64.deb ...\n",
            "Unpacking libqt5quick5:amd64 (5.15.3+dfsg-1) ...\n",
            "Selecting previously unselected package libqt5sensors5:amd64.\n",
            "Preparing to unpack .../26-libqt5sensors5_5.15.3-1_amd64.deb ...\n",
            "Unpacking libqt5sensors5:amd64 (5.15.3-1) ...\n",
            "Selecting previously unselected package libqt5webchannel5:amd64.\n",
            "Preparing to unpack .../27-libqt5webchannel5_5.15.3-1_amd64.deb ...\n",
            "Unpacking libqt5webchannel5:amd64 (5.15.3-1) ...\n",
            "Selecting previously unselected package libwoff1:amd64.\n",
            "Preparing to unpack .../28-libwoff1_1.0.2-1build4_amd64.deb ...\n",
            "Unpacking libwoff1:amd64 (1.0.2-1build4) ...\n",
            "Selecting previously unselected package libqt5webkit5:amd64.\n",
            "Preparing to unpack .../29-libqt5webkit5_5.212.0~alpha4-15ubuntu1_amd64.deb ...\n",
            "Unpacking libqt5webkit5:amd64 (5.212.0~alpha4-15ubuntu1) ...\n",
            "Selecting previously unselected package udev.\n",
            "Preparing to unpack .../30-udev_249.11-0ubuntu3.12_amd64.deb ...\n",
            "Unpacking udev (249.11-0ubuntu3.12) ...\n",
            "Selecting previously unselected package libavahi-glib1:amd64.\n",
            "Preparing to unpack .../31-libavahi-glib1_0.8-5ubuntu5.2_amd64.deb ...\n",
            "Unpacking libavahi-glib1:amd64 (0.8-5ubuntu5.2) ...\n",
            "Selecting previously unselected package libjson-glib-1.0-common.\n",
            "Preparing to unpack .../32-libjson-glib-1.0-common_1.6.6-1build1_all.deb ...\n",
            "Unpacking libjson-glib-1.0-common (1.6.6-1build1) ...\n",
            "Selecting previously unselected package libjson-glib-1.0-0:amd64.\n",
            "Preparing to unpack .../33-libjson-glib-1.0-0_1.6.6-1build1_amd64.deb ...\n",
            "Unpacking libjson-glib-1.0-0:amd64 (1.6.6-1build1) ...\n",
            "Selecting previously unselected package libmm-glib0:amd64.\n",
            "Preparing to unpack .../34-libmm-glib0_1.20.0-1~ubuntu22.04.4_amd64.deb ...\n",
            "Unpacking libmm-glib0:amd64 (1.20.0-1~ubuntu22.04.4) ...\n",
            "Selecting previously unselected package libnotify4:amd64.\n",
            "Preparing to unpack .../35-libnotify4_0.7.9-3ubuntu5.22.04.1_amd64.deb ...\n",
            "Unpacking libnotify4:amd64 (0.7.9-3ubuntu5.22.04.1) ...\n",
            "Selecting previously unselected package libproxy1v5:amd64.\n",
            "Preparing to unpack .../36-libproxy1v5_0.4.17-2_amd64.deb ...\n",
            "Unpacking libproxy1v5:amd64 (0.4.17-2) ...\n",
            "Selecting previously unselected package glib-networking-common.\n",
            "Preparing to unpack .../37-glib-networking-common_2.72.0-1_all.deb ...\n",
            "Unpacking glib-networking-common (2.72.0-1) ...\n",
            "Selecting previously unselected package glib-networking-services.\n",
            "Preparing to unpack .../38-glib-networking-services_2.72.0-1_amd64.deb ...\n",
            "Unpacking glib-networking-services (2.72.0-1) ...\n",
            "Selecting previously unselected package session-migration.\n",
            "Preparing to unpack .../39-session-migration_0.3.6_amd64.deb ...\n",
            "Unpacking session-migration (0.3.6) ...\n",
            "Selecting previously unselected package gsettings-desktop-schemas.\n",
            "Preparing to unpack .../40-gsettings-desktop-schemas_42.0-1ubuntu1_all.deb ...\n",
            "Unpacking gsettings-desktop-schemas (42.0-1ubuntu1) ...\n",
            "Selecting previously unselected package glib-networking:amd64.\n",
            "Preparing to unpack .../41-glib-networking_2.72.0-1_amd64.deb ...\n",
            "Unpacking glib-networking:amd64 (2.72.0-1) ...\n",
            "Selecting previously unselected package libsoup2.4-common.\n",
            "Preparing to unpack .../42-libsoup2.4-common_2.74.2-3ubuntu0.1_all.deb ...\n",
            "Unpacking libsoup2.4-common (2.74.2-3ubuntu0.1) ...\n",
            "Selecting previously unselected package libsoup2.4-1:amd64.\n",
            "Preparing to unpack .../43-libsoup2.4-1_2.74.2-3ubuntu0.1_amd64.deb ...\n",
            "Unpacking libsoup2.4-1:amd64 (2.74.2-3ubuntu0.1) ...\n",
            "Selecting previously unselected package geoclue-2.0.\n",
            "Preparing to unpack .../44-geoclue-2.0_2.5.7-3ubuntu3_amd64.deb ...\n",
            "Unpacking geoclue-2.0 (2.5.7-3ubuntu3) ...\n",
            "Selecting previously unselected package iio-sensor-proxy.\n",
            "Preparing to unpack .../45-iio-sensor-proxy_3.3-0ubuntu6_amd64.deb ...\n",
            "Unpacking iio-sensor-proxy (3.3-0ubuntu6) ...\n",
            "Selecting previously unselected package libfontenc1:amd64.\n",
            "Preparing to unpack .../46-libfontenc1_1%3a1.1.4-1build3_amd64.deb ...\n",
            "Unpacking libfontenc1:amd64 (1:1.1.4-1build3) ...\n",
            "Selecting previously unselected package libmbim-glib4:amd64.\n",
            "Preparing to unpack .../47-libmbim-glib4_1.28.0-1~ubuntu20.04.1_amd64.deb ...\n",
            "Unpacking libmbim-glib4:amd64 (1.28.0-1~ubuntu20.04.1) ...\n",
            "Selecting previously unselected package libmbim-proxy.\n",
            "Preparing to unpack .../48-libmbim-proxy_1.28.0-1~ubuntu20.04.1_amd64.deb ...\n",
            "Unpacking libmbim-proxy (1.28.0-1~ubuntu20.04.1) ...\n",
            "Selecting previously unselected package libnl-genl-3-200:amd64.\n",
            "Preparing to unpack .../49-libnl-genl-3-200_3.5.0-0.1_amd64.deb ...\n",
            "Unpacking libnl-genl-3-200:amd64 (3.5.0-0.1) ...\n",
            "Selecting previously unselected package libnss-mdns:amd64.\n",
            "Preparing to unpack .../50-libnss-mdns_0.15.1-1ubuntu1_amd64.deb ...\n",
            "Unpacking libnss-mdns:amd64 (0.15.1-1ubuntu1) ...\n",
            "Selecting previously unselected package libqmi-glib5:amd64.\n",
            "Preparing to unpack .../51-libqmi-glib5_1.32.0-1ubuntu0.22.04.1_amd64.deb ...\n",
            "Unpacking libqmi-glib5:amd64 (1.32.0-1ubuntu0.22.04.1) ...\n",
            "Selecting previously unselected package libqmi-proxy.\n",
            "Preparing to unpack .../52-libqmi-proxy_1.32.0-1ubuntu0.22.04.1_amd64.deb ...\n",
            "Unpacking libqmi-proxy (1.32.0-1ubuntu0.22.04.1) ...\n",
            "Selecting previously unselected package libwacom-bin.\n",
            "Preparing to unpack .../53-libwacom-bin_2.2.0-1_amd64.deb ...\n",
            "Unpacking libwacom-bin (2.2.0-1) ...\n",
            "Selecting previously unselected package libxfont2:amd64.\n",
            "Preparing to unpack .../54-libxfont2_1%3a2.0.5-1build1_amd64.deb ...\n",
            "Unpacking libxfont2:amd64 (1:2.0.5-1build1) ...\n",
            "Selecting previously unselected package libxkbfile1:amd64.\n",
            "Preparing to unpack .../55-libxkbfile1_1%3a1.1.0-1build3_amd64.deb ...\n",
            "Unpacking libxkbfile1:amd64 (1:1.1.0-1build3) ...\n",
            "Selecting previously unselected package modemmanager.\n",
            "Preparing to unpack .../56-modemmanager_1.20.0-1~ubuntu22.04.4_amd64.deb ...\n",
            "Unpacking modemmanager (1.20.0-1~ubuntu22.04.4) ...\n",
            "Selecting previously unselected package qt5-gtk-platformtheme:amd64.\n",
            "Preparing to unpack .../57-qt5-gtk-platformtheme_5.15.3+dfsg-2ubuntu0.2_amd64.deb ...\n",
            "Unpacking qt5-gtk-platformtheme:amd64 (5.15.3+dfsg-2ubuntu0.2) ...\n",
            "Selecting previously unselected package qttranslations5-l10n.\n",
            "Preparing to unpack .../58-qttranslations5-l10n_5.15.3-1_all.deb ...\n",
            "Unpacking qttranslations5-l10n (5.15.3-1) ...\n",
            "Selecting previously unselected package systemd-hwe-hwdb.\n",
            "Preparing to unpack .../59-systemd-hwe-hwdb_249.11.5_all.deb ...\n",
            "Unpacking systemd-hwe-hwdb (249.11.5) ...\n",
            "Selecting previously unselected package wpasupplicant.\n",
            "Preparing to unpack .../60-wpasupplicant_2%3a2.10-6ubuntu2.1_amd64.deb ...\n",
            "Unpacking wpasupplicant (2:2.10-6ubuntu2.1) ...\n",
            "Selecting previously unselected package x11-xkb-utils.\n",
            "Preparing to unpack .../61-x11-xkb-utils_7.7+5build4_amd64.deb ...\n",
            "Unpacking x11-xkb-utils (7.7+5build4) ...\n",
            "Selecting previously unselected package xfonts-encodings.\n",
            "Preparing to unpack .../62-xfonts-encodings_1%3a1.0.5-0ubuntu2_all.deb ...\n",
            "Unpacking xfonts-encodings (1:1.0.5-0ubuntu2) ...\n",
            "Selecting previously unselected package xfonts-utils.\n",
            "Preparing to unpack .../63-xfonts-utils_1%3a7.7+6build2_amd64.deb ...\n",
            "Unpacking xfonts-utils (1:7.7+6build2) ...\n",
            "Selecting previously unselected package xfonts-base.\n",
            "Preparing to unpack .../64-xfonts-base_1%3a1.0.5_all.deb ...\n",
            "Unpacking xfonts-base (1:1.0.5) ...\n",
            "Selecting previously unselected package xserver-common.\n",
            "Preparing to unpack .../65-xserver-common_2%3a21.1.4-2ubuntu1.7~22.04.12_all.deb ...\n",
            "Unpacking xserver-common (2:21.1.4-2ubuntu1.7~22.04.12) ...\n",
            "Selecting previously unselected package xnest.\n",
            "Preparing to unpack .../66-xnest_2%3a21.1.4-2ubuntu1.7~22.04.12_amd64.deb ...\n",
            "Unpacking xnest (2:21.1.4-2ubuntu1.7~22.04.12) ...\n",
            "Selecting previously unselected package usb-modeswitch-data.\n",
            "Preparing to unpack .../67-usb-modeswitch-data_20191128-4_all.deb ...\n",
            "Unpacking usb-modeswitch-data (20191128-4) ...\n",
            "Selecting previously unselected package usb-modeswitch.\n",
            "Preparing to unpack .../68-usb-modeswitch_2.6.1-3ubuntu2_amd64.deb ...\n",
            "Unpacking usb-modeswitch (2.6.1-3ubuntu2) ...\n",
            "Selecting previously unselected package wkhtmltopdf.\n",
            "Preparing to unpack .../69-wkhtmltopdf_0.12.6-2_amd64.deb ...\n",
            "Unpacking wkhtmltopdf (0.12.6-2) ...\n",
            "Setting up liblmdb0:amd64 (0.9.24-1build2) ...\n",
            "Setting up session-migration (0.3.6) ...\n",
            "Created symlink /etc/systemd/user/graphical-session-pre.target.wants/session-migration.service → /usr/lib/systemd/user/session-migration.service.\n",
            "Setting up libproxy1v5:amd64 (0.4.17-2) ...\n",
            "Setting up libxcb-xinput0:amd64 (1.14-3ubuntu3) ...\n",
            "Setting up libwoff1:amd64 (1.0.2-1build4) ...\n",
            "Setting up libhyphen0:amd64 (2.8.8-7build2) ...\n",
            "Setting up libmaxminddb0:amd64 (1.5.2-1build2) ...\n",
            "Setting up libxcb-keysyms1:amd64 (0.4.0-1build3) ...\n",
            "Setting up libxcb-render-util0:amd64 (0.3.9-1build3) ...\n",
            "Setting up libxcb-icccm4:amd64 (0.4.1-1.1build2) ...\n",
            "Setting up libxcb-util1:amd64 (0.4.0-1build2) ...\n",
            "Setting up libxcb-xkb1:amd64 (1.14-3ubuntu3) ...\n",
            "Setting up libxcb-image0:amd64 (0.4.0-2) ...\n",
            "Setting up libfontenc1:amd64 (1:1.1.4-1build3) ...\n",
            "Setting up libxcb-xinerama0:amd64 (1.14-3ubuntu3) ...\n",
            "Setting up qttranslations5-l10n (5.15.3-1) ...\n",
            "Setting up libnotify4:amd64 (0.7.9-3ubuntu5.22.04.1) ...\n",
            "Setting up xfonts-encodings (1:1.0.5-0ubuntu2) ...\n",
            "Setting up libxkbcommon-x11-0:amd64 (1.4.0-1) ...\n",
            "Setting up usb-modeswitch-data (20191128-4) ...\n",
            "Setting up udev (249.11-0ubuntu3.12) ...\n",
            "invoke-rc.d: could not determine current runlevel\n",
            "invoke-rc.d: policy-rc.d denied execution of start.\n",
            "Setting up libqt5core5a:amd64 (5.15.3+dfsg-2ubuntu0.2) ...\n",
            "Setting up libmtdev1:amd64 (1.1.6-1build4) ...\n",
            "Setting up libsoup2.4-common (2.74.2-3ubuntu0.1) ...\n",
            "Setting up systemd-hwe-hwdb (249.11.5) ...\n",
            "Setting up libmm-glib0:amd64 (1.20.0-1~ubuntu22.04.4) ...\n",
            "Setting up libqt5dbus5:amd64 (5.15.3+dfsg-2ubuntu0.2) ...\n",
            "Setting up libnl-genl-3-200:amd64 (3.5.0-0.1) ...\n",
            "Setting up libmd4c0:amd64 (0.4.8-1) ...\n",
            "Setting up libavahi-glib1:amd64 (0.8-5ubuntu5.2) ...\n",
            "Setting up libjson-glib-1.0-common (1.6.6-1build1) ...\n",
            "Setting up usb-modeswitch (2.6.1-3ubuntu2) ...\n",
            "Setting up libxkbfile1:amd64 (1:1.1.0-1build3) ...\n",
            "Setting up glib-networking-common (2.72.0-1) ...\n",
            "Setting up libqt5sensors5:amd64 (5.15.3-1) ...\n",
            "Setting up libdaemon0:amd64 (0.14-7.1ubuntu3) ...\n",
            "Setting up libavahi-core7:amd64 (0.8-5ubuntu5.2) ...\n",
            "Setting up libnss-mdns:amd64 (0.15.1-1ubuntu1) ...\n",
            "First installation detected...\n",
            "Checking NSS setup...\n",
            "Setting up libxfont2:amd64 (1:2.0.5-1build1) ...\n",
            "Setting up libevdev2:amd64 (1.12.1+dfsg-1) ...\n",
            "Setting up libgudev-1.0-0:amd64 (1:237-2build1) ...\n",
            "Setting up libmbim-glib4:amd64 (1.28.0-1~ubuntu20.04.1) ...\n",
            "Setting up libwacom-common (2.2.0-1) ...\n",
            "Setting up gsettings-desktop-schemas (42.0-1ubuntu1) ...\n",
            "Setting up glib-networking-services (2.72.0-1) ...\n",
            "Setting up iio-sensor-proxy (3.3-0ubuntu6) ...\n",
            "Setting up bind9-libs:amd64 (1:9.18.28-0ubuntu0.22.04.1) ...\n",
            "Setting up libwacom9:amd64 (2.2.0-1) ...\n",
            "Setting up x11-xkb-utils (7.7+5build4) ...\n",
            "Setting up libqt5positioning5:amd64 (5.15.3+dfsg-3) ...\n",
            "Setting up libmbim-proxy (1.28.0-1~ubuntu20.04.1) ...\n",
            "Setting up xfonts-utils (1:7.7+6build2) ...\n",
            "Setting up libqt5network5:amd64 (5.15.3+dfsg-2ubuntu0.2) ...\n",
            "Setting up libjson-glib-1.0-0:amd64 (1.6.6-1build1) ...\n",
            "Setting up libinput-bin (1.20.0-1ubuntu0.3) ...\n",
            "Setting up wpasupplicant (2:2.10-6ubuntu2.1) ...\n",
            "Created symlink /etc/systemd/system/dbus-fi.w1.wpa_supplicant1.service → /lib/systemd/system/wpa_supplicant.service.\n",
            "Created symlink /etc/systemd/system/multi-user.target.wants/wpa_supplicant.service → /lib/systemd/system/wpa_supplicant.service.\n",
            "Setting up xfonts-base (1:1.0.5) ...\n",
            "Setting up libqt5qml5:amd64 (5.15.3+dfsg-1) ...\n",
            "Setting up libqt5webchannel5:amd64 (5.15.3-1) ...\n",
            "Setting up libwacom-bin (2.2.0-1) ...\n",
            "Setting up xserver-common (2:21.1.4-2ubuntu1.7~22.04.12) ...\n",
            "Setting up bind9-host (1:9.18.28-0ubuntu0.22.04.1) ...\n",
            "Setting up libinput10:amd64 (1.20.0-1ubuntu0.3) ...\n",
            "Setting up libqt5qmlmodels5:amd64 (5.15.3+dfsg-1) ...\n",
            "Setting up libqt5gui5:amd64 (5.15.3+dfsg-2ubuntu0.2) ...\n",
            "Setting up libqmi-glib5:amd64 (1.32.0-1ubuntu0.22.04.1) ...\n",
            "Setting up libqt5widgets5:amd64 (5.15.3+dfsg-2ubuntu0.2) ...\n",
            "Setting up qt5-gtk-platformtheme:amd64 (5.15.3+dfsg-2ubuntu0.2) ...\n",
            "Setting up libqt5printsupport5:amd64 (5.15.3+dfsg-2ubuntu0.2) ...\n",
            "Setting up avahi-daemon (0.8-5ubuntu5.2) ...\n",
            "invoke-rc.d: could not determine current runlevel\n",
            "invoke-rc.d: policy-rc.d denied execution of force-reload.\n",
            "invoke-rc.d: could not determine current runlevel\n",
            "invoke-rc.d: policy-rc.d denied execution of start.\n",
            "Created symlink /etc/systemd/system/dbus-org.freedesktop.Avahi.service → /lib/systemd/system/avahi-daemon.service.\n",
            "Created symlink /etc/systemd/system/multi-user.target.wants/avahi-daemon.service → /lib/systemd/system/avahi-daemon.service.\n",
            "Created symlink /etc/systemd/system/sockets.target.wants/avahi-daemon.socket → /lib/systemd/system/avahi-daemon.socket.\n",
            "Setting up xnest (2:21.1.4-2ubuntu1.7~22.04.12) ...\n",
            "Setting up libqt5quick5:amd64 (5.15.3+dfsg-1) ...\n",
            "Setting up libqt5svg5:amd64 (5.15.3-1) ...\n",
            "Setting up libqmi-proxy (1.32.0-1ubuntu0.22.04.1) ...\n",
            "Setting up libqt5webkit5:amd64 (5.212.0~alpha4-15ubuntu1) ...\n",
            "Setting up modemmanager (1.20.0-1~ubuntu22.04.4) ...\n",
            "Created symlink /etc/systemd/system/dbus-org.freedesktop.ModemManager1.service → /lib/systemd/system/ModemManager.service.\n",
            "Created symlink /etc/systemd/system/multi-user.target.wants/ModemManager.service → /lib/systemd/system/ModemManager.service.\n",
            "Setting up wkhtmltopdf (0.12.6-2) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for dbus (1.12.20-2ubuntu4.1) ...\n",
            "Processing triggers for fontconfig (2.13.1-4.2ubuntu5) ...\n",
            "Processing triggers for hicolor-icon-theme (0.17-2) ...\n",
            "Processing triggers for libglib2.0-0:amd64 (2.72.4-0ubuntu2.4) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.4) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libumf.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n",
            "Setting up glib-networking:amd64 (2.72.0-1) ...\n",
            "Setting up libsoup2.4-1:amd64 (2.74.2-3ubuntu0.1) ...\n",
            "Setting up geoclue-2.0 (2.5.7-3ubuntu3) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.4) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libumf.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n",
            "Processing triggers for dbus (1.12.20-2ubuntu4.1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pyensembl install --release 93 --species human"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0s9FkJ1zJ3KO",
        "outputId": "5686c000-74fa-49ca-bd81-72918f6ba3dd"
      },
      "execution_count": 4,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "2025-01-03 07:53:55,736 - pyensembl.shell - INFO - Running 'install' for EnsemblRelease(release=93, species='homo_sapiens')\n",
            "2025-01-03 07:53:55,736 - pyensembl.download_cache - INFO - Fetching /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.93.gtf.gz from URL https://ftp.ensembl.org/pub/release-93/gtf/homo_sapiens/Homo_sapiens.GRCh38.93.gtf.gz\n",
            "2025-01-03 07:53:55,736 - datacache.download - INFO - Downloading https://ftp.ensembl.org/pub/release-93/gtf/homo_sapiens/Homo_sapiens.GRCh38.93.gtf.gz to /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.93.gtf.gz\n",
            "2025-01-03 07:54:47,154 - pyensembl.download_cache - INFO - Fetching /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.cdna.all.fa.gz from URL https://ftp.ensembl.org/pub/release-93/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
            "2025-01-03 07:54:47,154 - datacache.download - INFO - Downloading https://ftp.ensembl.org/pub/release-93/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz to /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
            "2025-01-03 07:56:09,031 - pyensembl.download_cache - INFO - Fetching /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.ncrna.fa.gz from URL https://ftp.ensembl.org/pub/release-93/fasta/homo_sapiens/ncrna/Homo_sapiens.GRCh38.ncrna.fa.gz\n",
            "2025-01-03 07:56:09,032 - datacache.download - INFO - Downloading https://ftp.ensembl.org/pub/release-93/fasta/homo_sapiens/ncrna/Homo_sapiens.GRCh38.ncrna.fa.gz to /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.ncrna.fa.gz\n",
            "2025-01-03 07:56:20,939 - pyensembl.download_cache - INFO - Fetching /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.pep.all.fa.gz from URL https://ftp.ensembl.org/pub/release-93/fasta/homo_sapiens/pep/Homo_sapiens.GRCh38.pep.all.fa.gz\n",
            "2025-01-03 07:56:20,939 - datacache.download - INFO - Downloading https://ftp.ensembl.org/pub/release-93/fasta/homo_sapiens/pep/Homo_sapiens.GRCh38.pep.all.fa.gz to /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.pep.all.fa.gz\n",
            "2025-01-03 07:56:38,055 - pyensembl.database - INFO - Creating database: /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.93.gtf.db\n",
            "2025-01-03 07:56:38,056 - pyensembl.database - INFO - Reading GTF from /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.93.gtf.gz\n",
            "2025-01-03 07:58:12,156 - datacache.database_helpers - INFO - Creating database /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.93.gtf.db containing: stop_codon, CDS, start_codon, exon, transcript, gene\n",
            "2025-01-03 07:58:12,156 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE stop_codon (end INT NOT NULL, source TEXT NOT NULL, exon_version TEXT NOT NULL, transcript_id TEXT NOT NULL, start INT NOT NULL, gene_biotype TEXT NOT NULL, feature TEXT NOT NULL, transcript_biotype TEXT NOT NULL, protein_version TEXT NOT NULL, exon_id TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_version TEXT NOT NULL, gene_id TEXT NOT NULL, ccds_id TEXT NOT NULL, strand TEXT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, transcript_support_level TEXT NOT NULL, seqname TEXT NOT NULL, gene_name TEXT NOT NULL, gene_version TEXT NOT NULL)\"\n",
            "2025-01-03 07:58:12,399 - datacache.database - INFO - Inserting 77440 rows into table stop_codon\n",
            "2025-01-03 07:58:12,832 - datacache.database - INFO - Creating index on stop_codon (seqname, start, end)\n",
            "2025-01-03 07:58:12,891 - datacache.database - INFO - Creating index on stop_codon (gene_name)\n",
            "2025-01-03 07:58:12,948 - datacache.database - INFO - Creating index on stop_codon (gene_id)\n",
            "2025-01-03 07:58:13,000 - datacache.database - INFO - Creating index on stop_codon (transcript_id)\n",
            "2025-01-03 07:58:13,057 - datacache.database - INFO - Creating index on stop_codon (transcript_name)\n",
            "2025-01-03 07:58:13,113 - datacache.database - INFO - Creating index on stop_codon (exon_id)\n",
            "2025-01-03 07:58:13,145 - datacache.database - INFO - Creating index on stop_codon (protein_id)\n",
            "2025-01-03 07:58:13,174 - datacache.database - INFO - Creating index on stop_codon (ccds_id)\n",
            "2025-01-03 07:58:13,226 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE CDS (end INT NOT NULL, source TEXT NOT NULL, exon_version TEXT NOT NULL, transcript_id TEXT NOT NULL, start INT NOT NULL, gene_biotype TEXT NOT NULL, feature TEXT NOT NULL, transcript_biotype TEXT NOT NULL, protein_version TEXT NOT NULL, exon_id TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_version TEXT NOT NULL, gene_id TEXT NOT NULL, ccds_id TEXT NOT NULL, strand TEXT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, transcript_support_level TEXT NOT NULL, seqname TEXT NOT NULL, gene_name TEXT NOT NULL, gene_version TEXT NOT NULL)\"\n",
            "2025-01-03 07:58:15,025 - datacache.database - INFO - Inserting 734649 rows into table CDS\n",
            "2025-01-03 07:58:20,229 - datacache.database - INFO - Creating index on CDS (seqname, start, end)\n",
            "2025-01-03 07:58:20,891 - datacache.database - INFO - Creating index on CDS (gene_name)\n",
            "2025-01-03 07:58:21,403 - datacache.database - INFO - Creating index on CDS (gene_id)\n",
            "2025-01-03 07:58:21,972 - datacache.database - INFO - Creating index on CDS (transcript_id)\n",
            "2025-01-03 07:58:22,520 - datacache.database - INFO - Creating index on CDS (transcript_name)\n",
            "2025-01-03 07:58:23,055 - datacache.database - INFO - Creating index on CDS (exon_id)\n",
            "2025-01-03 07:58:23,416 - datacache.database - INFO - Creating index on CDS (protein_id)\n",
            "2025-01-03 07:58:23,948 - datacache.database - INFO - Creating index on CDS (ccds_id)\n",
            "2025-01-03 07:58:24,417 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE start_codon (end INT NOT NULL, source TEXT NOT NULL, exon_version TEXT NOT NULL, transcript_id TEXT NOT NULL, start INT NOT NULL, gene_biotype TEXT NOT NULL, feature TEXT NOT NULL, transcript_biotype TEXT NOT NULL, protein_version TEXT NOT NULL, exon_id TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_version TEXT NOT NULL, gene_id TEXT NOT NULL, ccds_id TEXT NOT NULL, strand TEXT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, transcript_support_level TEXT NOT NULL, seqname TEXT NOT NULL, gene_name TEXT NOT NULL, gene_version TEXT NOT NULL)\"\n",
            "2025-01-03 07:58:24,626 - datacache.database - INFO - Inserting 85453 rows into table start_codon\n",
            "2025-01-03 07:58:25,091 - datacache.database - INFO - Creating index on start_codon (seqname, start, end)\n",
            "2025-01-03 07:58:25,154 - datacache.database - INFO - Creating index on start_codon (gene_name)\n",
            "2025-01-03 07:58:25,212 - datacache.database - INFO - Creating index on start_codon (gene_id)\n",
            "2025-01-03 07:58:25,270 - datacache.database - INFO - Creating index on start_codon (transcript_id)\n",
            "2025-01-03 07:58:25,338 - datacache.database - INFO - Creating index on start_codon (transcript_name)\n",
            "2025-01-03 07:58:25,398 - datacache.database - INFO - Creating index on start_codon (exon_id)\n",
            "2025-01-03 07:58:25,438 - datacache.database - INFO - Creating index on start_codon (protein_id)\n",
            "2025-01-03 07:58:25,477 - datacache.database - INFO - Creating index on start_codon (ccds_id)\n",
            "2025-01-03 07:58:25,530 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE exon (end INT NOT NULL, source TEXT NOT NULL, exon_version TEXT NOT NULL, transcript_id TEXT NOT NULL, start INT NOT NULL, gene_biotype TEXT NOT NULL, feature TEXT NOT NULL, transcript_biotype TEXT NOT NULL, protein_version TEXT NOT NULL, exon_id TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_version TEXT NOT NULL, gene_id TEXT NOT NULL, ccds_id TEXT NOT NULL, strand TEXT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, transcript_support_level TEXT NOT NULL, seqname TEXT NOT NULL, gene_name TEXT NOT NULL, gene_version TEXT NOT NULL)\"\n",
            "2025-01-03 07:58:28,456 - datacache.database - INFO - Inserting 1237303 rows into table exon\n",
            "2025-01-03 07:58:36,369 - datacache.database - INFO - Creating index on exon (seqname, start, end)\n",
            "2025-01-03 07:58:37,660 - datacache.database - INFO - Creating index on exon (gene_name)\n",
            "2025-01-03 07:58:38,671 - datacache.database - INFO - Creating index on exon (gene_id)\n",
            "2025-01-03 07:58:39,659 - datacache.database - INFO - Creating index on exon (transcript_id)\n",
            "2025-01-03 07:58:40,670 - datacache.database - INFO - Creating index on exon (transcript_name)\n",
            "2025-01-03 07:58:41,754 - datacache.database - INFO - Creating index on exon (exon_id)\n",
            "2025-01-03 07:58:42,955 - datacache.database - INFO - Creating index on exon (protein_id)\n",
            "2025-01-03 07:58:44,038 - datacache.database - INFO - Creating index on exon (ccds_id)\n",
            "2025-01-03 07:58:45,616 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE transcript (end INT NOT NULL, source TEXT NOT NULL, exon_version TEXT NOT NULL, transcript_id TEXT UNIQUE PRIMARY KEY NOT NULL, start INT NOT NULL, gene_biotype TEXT NOT NULL, feature TEXT NOT NULL, transcript_biotype TEXT NOT NULL, protein_version TEXT NOT NULL, exon_id TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_version TEXT NOT NULL, gene_id TEXT NOT NULL, ccds_id TEXT NOT NULL, strand TEXT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, transcript_support_level TEXT NOT NULL, seqname TEXT NOT NULL, gene_name TEXT NOT NULL, gene_version TEXT NOT NULL)\"\n",
            "2025-01-03 07:58:46,349 - datacache.database - INFO - Inserting 203742 rows into table transcript\n",
            "2025-01-03 07:58:47,882 - datacache.database - INFO - Creating index on transcript (seqname, start, end)\n",
            "2025-01-03 07:58:48,030 - datacache.database - INFO - Creating index on transcript (gene_name)\n",
            "2025-01-03 07:58:48,180 - datacache.database - INFO - Creating index on transcript (gene_id)\n",
            "2025-01-03 07:58:48,330 - datacache.database - INFO - Creating index on transcript (transcript_name)\n",
            "2025-01-03 07:58:48,543 - datacache.database - INFO - Creating index on transcript (exon_id)\n",
            "2025-01-03 07:58:48,645 - datacache.database - INFO - Creating index on transcript (protein_id)\n",
            "2025-01-03 07:58:48,747 - datacache.database - INFO - Creating index on transcript (ccds_id)\n",
            "2025-01-03 07:58:48,894 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE gene (end INT NOT NULL, source TEXT NOT NULL, exon_version TEXT NOT NULL, transcript_id TEXT NOT NULL, start INT NOT NULL, gene_biotype TEXT NOT NULL, feature TEXT NOT NULL, transcript_biotype TEXT NOT NULL, protein_version TEXT NOT NULL, exon_id TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_version TEXT NOT NULL, gene_id TEXT UNIQUE PRIMARY KEY NOT NULL, ccds_id TEXT NOT NULL, strand TEXT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, transcript_support_level TEXT NOT NULL, seqname TEXT NOT NULL, gene_name TEXT NOT NULL, gene_version TEXT NOT NULL)\"\n",
            "2025-01-03 07:58:49,023 - datacache.database - INFO - Inserting 58395 rows into table gene\n",
            "2025-01-03 07:58:49,408 - datacache.database - INFO - Creating index on gene (seqname, start, end)\n",
            "2025-01-03 07:58:49,449 - datacache.database - INFO - Creating index on gene (gene_name)\n",
            "2025-01-03 07:58:49,490 - datacache.database - INFO - Creating index on gene (transcript_id)\n",
            "2025-01-03 07:58:49,513 - datacache.database - INFO - Creating index on gene (transcript_name)\n",
            "2025-01-03 07:58:49,534 - datacache.database - INFO - Creating index on gene (exon_id)\n",
            "2025-01-03 07:58:49,555 - datacache.database - INFO - Creating index on gene (protein_id)\n",
            "2025-01-03 07:58:49,576 - datacache.database - INFO - Creating index on gene (ccds_id)\n",
            "2025-01-03 07:58:49,597 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE _datacache_metadata (version INT)\"\n",
            "2025-01-03 07:58:49,597 - datacache.database - INFO - Running sqlite query: \"INSERT INTO _datacache_metadata VALUES (3)\"\n",
            "2025-01-03 07:58:49,899 - pyensembl.sequence_data - INFO - Parsing sequences from FASTA file at /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
            "2025-01-03 07:58:57,367 - pyensembl.sequence_data - INFO - Saving sequence dictionary to /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.cdna.all.fa.gz.pickle\n",
            "2025-01-03 07:58:58,646 - pyensembl.sequence_data - INFO - Parsing sequences from FASTA file at /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.ncrna.fa.gz\n",
            "2025-01-03 07:58:59,755 - pyensembl.sequence_data - INFO - Saving sequence dictionary to /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.ncrna.fa.gz.pickle\n",
            "2025-01-03 07:58:59,811 - pyensembl.sequence_data - INFO - Parsing sequences from FASTA file at /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.pep.all.fa.gz\n",
            "2025-01-03 07:59:01,202 - pyensembl.sequence_data - INFO - Saving sequence dictionary to /root/.cache/pyensembl/GRCh38/ensembl93/Homo_sapiens.GRCh38.pep.all.fa.gz.pickle\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!git clone https://github.com/SysteMHC/SysteMHC_Binaries.git"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "oBr9X0wKLkit",
        "outputId": "bf1bdeb6-0092-49bd-d78a-5127f2140a4b"
      },
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'SysteMHC_Binaries'...\n",
            "remote: Enumerating objects: 12968, done.\u001b[K\n",
            "remote: Total 12968 (delta 0), reused 0 (delta 0), pack-reused 12968 (from 1)\u001b[K\n",
            "Receiving objects: 100% (12968/12968), 276.77 MiB | 22.25 MiB/s, done.\n",
            "Resolving deltas: 100% (885/885), done.\n",
            "Updating files: 100% (11756/11756), done.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%cd SysteMHC_Binaries\n",
        "%cd netMHC"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iipMfyVDLvu2",
        "outputId": "56ca73b1-4454-45bb-b9ff-e99500977573"
      },
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[Errno 2] No such file or directory: 'SysteMHC_Binaries'\n",
            "/content/vaxrank/SysteMHC_Binaries\n",
            "/content/vaxrank/SysteMHC_Binaries/netMHC\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://raw.githubusercontent.com/openvax/vaxrank/main/tests/data/b16.f10/b16.vcf -P vaxrank/tests/data/b16.f10/\n",
        "!wget https://raw.githubusercontent.com/openvax/vaxrank/main/tests/data/b16.f10/b16.combined.bam -P vaxrank/tests/data/b16.f10/\n",
        "\n",
        "!cp vaxrank/tests/data/b16.f10/b16.vcf mutations.vcf\n",
        "!cp vaxrank/tests/data/b16.f10/b16.combined.bam tumor.bam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "VtXi4TJ1LTLY",
        "outputId": "32d91714-1b06-4ce3-a04f-b05c81758718"
      },
      "execution_count": 24,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-01-03 08:28:59--  https://raw.githubusercontent.com/openvax/vaxrank/main/tests/data/b16.f10/b16.vcf\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.110.133, 185.199.108.133, 185.199.109.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.110.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 186 [text/plain]\n",
            "Saving to: ‘vaxrank/tests/data/b16.f10/b16.vcf.2’\n",
            "\n",
            "\rb16.vcf.2             0%[                    ]       0  --.-KB/s               \rb16.vcf.2           100%[===================>]     186  --.-KB/s    in 0s      \n",
            "\n",
            "2025-01-03 08:28:59 (12.6 MB/s) - ‘vaxrank/tests/data/b16.f10/b16.vcf.2’ saved [186/186]\n",
            "\n",
            "--2025-01-03 08:28:59--  https://raw.githubusercontent.com/openvax/vaxrank/main/tests/data/b16.f10/b16.combined.bam\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.111.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 16103 (16K) [application/octet-stream]\n",
            "Saving to: ‘vaxrank/tests/data/b16.f10/b16.combined.bam.2’\n",
            "\n",
            "b16.combined.bam.2  100%[===================>]  15.73K  --.-KB/s    in 0.004s  \n",
            "\n",
            "2025-01-03 08:28:59 (3.50 MB/s) - ‘vaxrank/tests/data/b16.f10/b16.combined.bam.2’ saved [16103/16103]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install mhctools"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FRVptJN9LcBQ",
        "outputId": "60c3f431-afca-434c-9908-cf44966b1dee"
      },
      "execution_count": 92,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: mhctools in /usr/local/lib/python3.10/dist-packages (1.9.0)\n",
            "Requirement already satisfied: numpy>=1.7 in /usr/local/lib/python3.10/dist-packages (from mhctools) (1.26.4)\n",
            "Requirement already satisfied: pandas>=0.13.1 in /usr/local/lib/python3.10/dist-packages (from mhctools) (2.2.2)\n",
            "Requirement already satisfied: varcode>=0.5.9 in /usr/local/lib/python3.10/dist-packages (from mhctools) (1.2.1)\n",
            "Requirement already satisfied: pyensembl>=1.0.3 in /usr/local/lib/python3.10/dist-packages (from mhctools) (2.3.13)\n",
            "Requirement already satisfied: sercol>=0.0.2 in /usr/local/lib/python3.10/dist-packages (from mhctools) (1.0.0)\n",
            "Requirement already satisfied: mhcflurry>=2.0.0 in /usr/local/lib/python3.10/dist-packages (from mhctools) (2.1.4)\n",
            "Requirement already satisfied: mhcnames>=0.3.2 in /usr/local/lib/python3.10/dist-packages (from mhctools) (0.4.8)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools) (1.17.0)\n",
            "Requirement already satisfied: appdirs in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools) (1.4.4)\n",
            "Requirement already satisfied: scikit-learn in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools) (1.6.0)\n",
            "Requirement already satisfied: mhcgnomes>=0.8.4 in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools) (1.8.6)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools) (6.0.2)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools) (4.67.1)\n",
            "Requirement already satisfied: tensorflow>=2.15.0 in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools) (2.17.1)\n",
            "Requirement already satisfied: tf-keras in /usr/local/lib/python3.10/dist-packages (from mhcflurry>=2.0.0->mhctools) (2.17.0)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.10/dist-packages (from pandas>=0.13.1->mhctools) (2.8.2)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas>=0.13.1->mhctools) (2024.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.10/dist-packages (from pandas>=0.13.1->mhctools) (2024.2)\n",
            "Requirement already satisfied: typechecks<1.0.0,>=0.0.2 in /usr/local/lib/python3.10/dist-packages (from pyensembl>=1.0.3->mhctools) (0.1.0)\n",
            "Requirement already satisfied: datacache<2.0.0,>=1.4.0 in /usr/local/lib/python3.10/dist-packages (from pyensembl>=1.0.3->mhctools) (1.4.1)\n",
            "Requirement already satisfied: memoized-property>=1.0.2 in /usr/local/lib/python3.10/dist-packages (from pyensembl>=1.0.3->mhctools) (1.0.3)\n",
            "Requirement already satisfied: tinytimer<1.0.0,>=0.0.0 in /usr/local/lib/python3.10/dist-packages (from pyensembl>=1.0.3->mhctools) (0.0.0)\n",
            "Requirement already satisfied: gtfparse<3.0.0,>=2.5.0 in /usr/local/lib/python3.10/dist-packages (from pyensembl>=1.0.3->mhctools) (2.5.0)\n",
            "Requirement already satisfied: serializable<1.0.0,>=0.2.1 in /usr/local/lib/python3.10/dist-packages (from pyensembl>=1.0.3->mhctools) (0.4.1)\n",
            "Requirement already satisfied: pylint<3.0.0,>=2.17.2 in /usr/local/lib/python3.10/dist-packages (from pyensembl>=1.0.3->mhctools) (2.17.7)\n",
            "Requirement already satisfied: simplejson in /usr/local/lib/python3.10/dist-packages (from sercol>=0.0.2->mhctools) (3.19.3)\n",
            "Requirement already satisfied: biopython>=1.64 in /usr/local/lib/python3.10/dist-packages (from varcode>=0.5.9->mhctools) (1.84)\n",
            "Requirement already satisfied: pyvcf3>=1.0.0 in /usr/local/lib/python3.10/dist-packages (from varcode>=0.5.9->mhctools) (1.0.3)\n",
            "Requirement already satisfied: progressbar33>=2.4 in /usr/local/lib/python3.10/dist-packages (from datacache<2.0.0,>=1.4.0->pyensembl>=1.0.3->mhctools) (2.4)\n",
            "Requirement already satisfied: requests>=2.5.1 in /usr/local/lib/python3.10/dist-packages (from datacache<2.0.0,>=1.4.0->pyensembl>=1.0.3->mhctools) (2.32.3)\n",
            "Requirement already satisfied: mock in /usr/local/lib/python3.10/dist-packages (from datacache<2.0.0,>=1.4.0->pyensembl>=1.0.3->mhctools) (5.1.0)\n",
            "Requirement already satisfied: polars<0.21.0,>=0.20.2 in /usr/local/lib/python3.10/dist-packages (from gtfparse<3.0.0,>=2.5.0->pyensembl>=1.0.3->mhctools) (0.20.31)\n",
            "Requirement already satisfied: pyarrow<14.1.0,>=14.0.2 in /usr/local/lib/python3.10/dist-packages (from gtfparse<3.0.0,>=2.5.0->pyensembl>=1.0.3->mhctools) (14.0.2)\n",
            "Requirement already satisfied: platformdirs>=2.2.0 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl>=1.0.3->mhctools) (4.3.6)\n",
            "Requirement already satisfied: astroid<=2.17.0-dev0,>=2.15.8 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl>=1.0.3->mhctools) (2.15.8)\n",
            "Requirement already satisfied: isort<6,>=4.2.5 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl>=1.0.3->mhctools) (5.13.2)\n",
            "Requirement already satisfied: mccabe<0.8,>=0.6 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl>=1.0.3->mhctools) (0.7.0)\n",
            "Requirement already satisfied: tomlkit>=0.10.1 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl>=1.0.3->mhctools) (0.13.2)\n",
            "Requirement already satisfied: dill>=0.2 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl>=1.0.3->mhctools) (0.3.9)\n",
            "Requirement already satisfied: tomli>=1.1.0 in /usr/local/lib/python3.10/dist-packages (from pylint<3.0.0,>=2.17.2->pyensembl>=1.0.3->mhctools) (2.2.1)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.10/dist-packages (from pyvcf3>=1.0.0->varcode>=0.5.9->mhctools) (75.1.0)\n",
            "Requirement already satisfied: absl-py>=1.0.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (1.4.0)\n",
            "Requirement already satisfied: astunparse>=1.6.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (1.6.3)\n",
            "Requirement already satisfied: flatbuffers>=24.3.25 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (24.3.25)\n",
            "Requirement already satisfied: gast!=0.5.0,!=0.5.1,!=0.5.2,>=0.2.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (0.6.0)\n",
            "Requirement already satisfied: google-pasta>=0.1.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (0.2.0)\n",
            "Requirement already satisfied: h5py>=3.10.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (3.12.1)\n",
            "Requirement already satisfied: libclang>=13.0.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (18.1.1)\n",
            "Requirement already satisfied: ml-dtypes<0.5.0,>=0.3.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (0.4.1)\n",
            "Requirement already satisfied: opt-einsum>=2.3.2 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (3.4.0)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (24.2)\n",
            "Requirement already satisfied: protobuf!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0dev,>=3.20.3 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (4.25.5)\n",
            "Requirement already satisfied: termcolor>=1.1.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (2.5.0)\n",
            "Requirement already satisfied: typing-extensions>=3.6.6 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (4.12.2)\n",
            "Requirement already satisfied: wrapt>=1.11.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (1.17.0)\n",
            "Requirement already satisfied: grpcio<2.0,>=1.24.3 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (1.68.1)\n",
            "Requirement already satisfied: tensorboard<2.18,>=2.17 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (2.17.1)\n",
            "Requirement already satisfied: keras>=3.2.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (3.5.0)\n",
            "Requirement already satisfied: tensorflow-io-gcs-filesystem>=0.23.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (0.37.1)\n",
            "Requirement already satisfied: scipy>=1.6.0 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->mhcflurry>=2.0.0->mhctools) (1.13.1)\n",
            "Requirement already satisfied: joblib>=1.2.0 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->mhcflurry>=2.0.0->mhctools) (1.4.2)\n",
            "Requirement already satisfied: threadpoolctl>=3.1.0 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->mhcflurry>=2.0.0->mhctools) (3.5.0)\n",
            "Requirement already satisfied: lazy-object-proxy>=1.4.0 in /usr/local/lib/python3.10/dist-packages (from astroid<=2.17.0-dev0,>=2.15.8->pylint<3.0.0,>=2.17.2->pyensembl>=1.0.3->mhctools) (1.10.0)\n",
            "Requirement already satisfied: wheel<1.0,>=0.23.0 in /usr/local/lib/python3.10/dist-packages (from astunparse>=1.6.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (0.45.1)\n",
            "Requirement already satisfied: rich in /usr/local/lib/python3.10/dist-packages (from keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (13.9.4)\n",
            "Requirement already satisfied: namex in /usr/local/lib/python3.10/dist-packages (from keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (0.0.8)\n",
            "Requirement already satisfied: optree in /usr/local/lib/python3.10/dist-packages (from keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (0.13.1)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests>=2.5.1->datacache<2.0.0,>=1.4.0->pyensembl>=1.0.3->mhctools) (3.4.0)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests>=2.5.1->datacache<2.0.0,>=1.4.0->pyensembl>=1.0.3->mhctools) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests>=2.5.1->datacache<2.0.0,>=1.4.0->pyensembl>=1.0.3->mhctools) (2.2.3)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests>=2.5.1->datacache<2.0.0,>=1.4.0->pyensembl>=1.0.3->mhctools) (2024.12.14)\n",
            "Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.10/dist-packages (from tensorboard<2.18,>=2.17->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (3.7)\n",
            "Requirement already satisfied: tensorboard-data-server<0.8.0,>=0.7.0 in /usr/local/lib/python3.10/dist-packages (from tensorboard<2.18,>=2.17->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (0.7.2)\n",
            "Requirement already satisfied: werkzeug>=1.0.1 in /usr/local/lib/python3.10/dist-packages (from tensorboard<2.18,>=2.17->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (3.1.3)\n",
            "Requirement already satisfied: MarkupSafe>=2.1.1 in /usr/local/lib/python3.10/dist-packages (from werkzeug>=1.0.1->tensorboard<2.18,>=2.17->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (3.0.2)\n",
            "Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.10/dist-packages (from rich->keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (3.0.0)\n",
            "Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /usr/local/lib/python3.10/dist-packages (from rich->keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (2.18.0)\n",
            "Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.10/dist-packages (from markdown-it-py>=2.2.0->rich->keras>=3.2.0->tensorflow>=2.15.0->mhcflurry>=2.0.0->mhctools) (0.1.2)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!vaxrank \\\n",
        "    --vcf /content/vaxrank/mutations.vcf \\\n",
        "    --bam /content/vaxrank/tumor.bam \\\n",
        "    --vaccine-peptide-length 25 \\\n",
        "    --mhc-predictor netmhc \\\n",
        "    --mhc-alleles H2-Kb,H2-Db \\\n",
        "    --padding-around-mutation 5 \\\n",
        "    --output-ascii-report vaccine-peptides.txt \\\n",
        "    --output-pdf-report vaccine-peptides.pdf \\\n",
        "    --output-html-report vaccine-peptides.html\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "VktAT58qMO12",
        "outputId": "d18b0b50-aa06-417f-c299-5ec78290f591"
      },
      "execution_count": 93,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "2025-01-03 09:31:03,310 - vaxrank.cli:407 - INFO - Namespace(vcf=['/content/vaxrank/mutations.vcf'], maf=[], variant=[], genome=None, download_reference_genome_data=False, json_variants=[], bam='/content/vaxrank/tumor.bam', min_mapping_quality=1, use_duplicate_reads=False, drop_secondary_alignments=False, num_rna_decompression_threads=1, min_variant_sequence_coverage=2, variant_sequence_assembly=True, reference_context_size=30, protein_sequence_length=20, max_reference_transcript_mismatches=2, count_mismatches_after_variant=False, min_transcript_prefix_length=10, max_protein_sequences_per_variant=1, min_alt_rna_reads=3, min_alt_rna_fragments=2, min_alt_rna_fraction=0.005, min_ratio_alt_to_other_fragments=3.0, mhc_predictor='netmhc', mhc_predictor_path=None, mhc_predictor_models_path=None, mhc_peptide_lengths=None, mhc_epitope_lengths=None, mhc_alleles_file=None, mhc_alleles='H2-Kb,H2-Db', do_not_raise_on_error=False, vaccine_peptide_length=25, padding_around_mutation=5, max_vaccine_peptides_per_mutation=1, min_epitope_score=1e-10, num_epitopes_per_vaccine_peptide=None, output_patient_id='', output_csv='', output_ascii_report='vaccine-peptides.txt', output_html_report='vaccine-peptides.html', output_pdf_report='vaccine-peptides.pdf', output_json_file='', output_xlsx_report='', output_neoepitope_report='', output_reviewed_by='', output_final_review='', output_passing_variants_csv='', output_isovar_csv='', log_path='python.log', max_mutations_in_report=None, manufacturability=True, wt_epitopes=True, cosmic_vcf_filename='')\n",
            "2025-01-03 09:31:03,310 - vaxrank.cli:320 - INFO - MHC alleles: ['H-2-Kb', 'H-2-Db']\n",
            "2025-01-03 09:31:03,320 - vaxrank.cli:323 - INFO - Variants: <VariantCollection from '/content/vaxrank/mutations.vcf' with 5 elements>\n",
            "  -- Variant(contig='11', start=101177505, ref='T', alt='C', reference_name='GRCm38')\n",
            "  -- Variant(contig='13', start=5864876, ref='', alt='G', reference_name='GRCm38')\n",
            "  -- Variant(contig='4', start=45802539, ref='G', alt='C', reference_name='GRCm38')\n",
            "  -- Variant(contig='9', start=82927102, ref='G', alt='T', reference_name='GRCm38')\n",
            "  -- Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38')\n",
            "2025-01-03 09:31:03,320 - mhctools.cli.args:179 - INFO - Building MHC binding prediction function for alleles ['H-2-Kb', 'H-2-Db'] and epitope lengths None\n",
            "Traceback (most recent call last):\n",
            "  File \"/usr/local/bin/vaxrank\", line 8, in <module>\n",
            "    sys.exit(main())\n",
            "  File \"/usr/local/lib/python3.10/dist-packages/vaxrank/cli.py\", line 410, in main\n",
            "    data = ranked_vaccine_peptides_with_metadata_from_parsed_args(args)\n",
            "  File \"/usr/local/lib/python3.10/dist-packages/vaxrank/cli.py\", line 325, in ranked_vaccine_peptides_with_metadata_from_parsed_args\n",
            "    vaxrank_results = run_vaxrank_from_parsed_args(args)\n",
            "  File \"/usr/local/lib/python3.10/dist-packages/vaxrank/cli.py\", line 269, in run_vaxrank_from_parsed_args\n",
            "    mhc_predictor = mhc_binding_predictor_from_args(args)\n",
            "  File \"/usr/local/lib/python3.10/dist-packages/mhctools/cli/args.py\", line 202, in mhc_binding_predictor_from_args\n",
            "    return mhc_class(**kwargs)\n",
            "  File \"/usr/local/lib/python3.10/dist-packages/mhctools/netmhc.py\", line 30, in NetMHC\n",
            "    help_output = check_output([program_name, \"-h\"], stderr=devnull)\n",
            "  File \"/usr/lib/python3.10/subprocess.py\", line 421, in check_output\n",
            "    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,\n",
            "  File \"/usr/lib/python3.10/subprocess.py\", line 503, in run\n",
            "    with Popen(*popenargs, **kwargs) as process:\n",
            "  File \"/usr/lib/python3.10/subprocess.py\", line 971, in __init__\n",
            "    self._execute_child(args, executable, preexec_fn, close_fds,\n",
            "  File \"/usr/lib/python3.10/subprocess.py\", line 1863, in _execute_child\n",
            "    raise child_exception_type(errno_num, err_msg, err_filename)\n",
            "FileNotFoundError: [Errno 2] No such file or directory: 'netMHC'\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install mhcflurry\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "GZBPXbQLgmE3",
        "outputId": "25f1ffcf-a852-4cdd-acef-90a27f7fa935"
      },
      "execution_count": 95,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: mhcflurry in /usr/local/lib/python3.10/dist-packages (2.1.4)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.10/dist-packages (from mhcflurry) (1.17.0)\n",
            "Requirement already satisfied: pandas>=0.20.3 in /usr/local/lib/python3.10/dist-packages (from mhcflurry) (2.2.2)\n",
            "Requirement already satisfied: appdirs in /usr/local/lib/python3.10/dist-packages (from mhcflurry) (1.4.4)\n",
            "Requirement already satisfied: scikit-learn in /usr/local/lib/python3.10/dist-packages (from mhcflurry) (1.6.0)\n",
            "Requirement already satisfied: mhcgnomes>=0.8.4 in /usr/local/lib/python3.10/dist-packages (from mhcflurry) (1.8.6)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.10/dist-packages (from mhcflurry) (6.0.2)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.10/dist-packages (from mhcflurry) (4.67.1)\n",
            "Requirement already satisfied: tensorflow>=2.15.0 in /usr/local/lib/python3.10/dist-packages (from mhcflurry) (2.17.1)\n",
            "Requirement already satisfied: tf-keras in /usr/local/lib/python3.10/dist-packages (from mhcflurry) (2.17.0)\n",
            "Requirement already satisfied: serializable in /usr/local/lib/python3.10/dist-packages (from mhcgnomes>=0.8.4->mhcflurry) (0.4.1)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from mhcgnomes>=0.8.4->mhcflurry) (1.26.4)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.10/dist-packages (from pandas>=0.20.3->mhcflurry) (2.8.2)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas>=0.20.3->mhcflurry) (2024.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.10/dist-packages (from pandas>=0.20.3->mhcflurry) (2024.2)\n",
            "Requirement already satisfied: absl-py>=1.0.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (1.4.0)\n",
            "Requirement already satisfied: astunparse>=1.6.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (1.6.3)\n",
            "Requirement already satisfied: flatbuffers>=24.3.25 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (24.3.25)\n",
            "Requirement already satisfied: gast!=0.5.0,!=0.5.1,!=0.5.2,>=0.2.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (0.6.0)\n",
            "Requirement already satisfied: google-pasta>=0.1.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (0.2.0)\n",
            "Requirement already satisfied: h5py>=3.10.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (3.12.1)\n",
            "Requirement already satisfied: libclang>=13.0.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (18.1.1)\n",
            "Requirement already satisfied: ml-dtypes<0.5.0,>=0.3.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (0.4.1)\n",
            "Requirement already satisfied: opt-einsum>=2.3.2 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (3.4.0)\n",
            "Requirement already satisfied: packaging in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (24.2)\n",
            "Requirement already satisfied: protobuf!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0dev,>=3.20.3 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (4.25.5)\n",
            "Requirement already satisfied: requests<3,>=2.21.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (2.32.3)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (75.1.0)\n",
            "Requirement already satisfied: termcolor>=1.1.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (2.5.0)\n",
            "Requirement already satisfied: typing-extensions>=3.6.6 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (4.12.2)\n",
            "Requirement already satisfied: wrapt>=1.11.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (1.17.0)\n",
            "Requirement already satisfied: grpcio<2.0,>=1.24.3 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (1.68.1)\n",
            "Requirement already satisfied: tensorboard<2.18,>=2.17 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (2.17.1)\n",
            "Requirement already satisfied: keras>=3.2.0 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (3.5.0)\n",
            "Requirement already satisfied: tensorflow-io-gcs-filesystem>=0.23.1 in /usr/local/lib/python3.10/dist-packages (from tensorflow>=2.15.0->mhcflurry) (0.37.1)\n",
            "Requirement already satisfied: scipy>=1.6.0 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->mhcflurry) (1.13.1)\n",
            "Requirement already satisfied: joblib>=1.2.0 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->mhcflurry) (1.4.2)\n",
            "Requirement already satisfied: threadpoolctl>=3.1.0 in /usr/local/lib/python3.10/dist-packages (from scikit-learn->mhcflurry) (3.5.0)\n",
            "Requirement already satisfied: wheel<1.0,>=0.23.0 in /usr/local/lib/python3.10/dist-packages (from astunparse>=1.6.0->tensorflow>=2.15.0->mhcflurry) (0.45.1)\n",
            "Requirement already satisfied: rich in /usr/local/lib/python3.10/dist-packages (from keras>=3.2.0->tensorflow>=2.15.0->mhcflurry) (13.9.4)\n",
            "Requirement already satisfied: namex in /usr/local/lib/python3.10/dist-packages (from keras>=3.2.0->tensorflow>=2.15.0->mhcflurry) (0.0.8)\n",
            "Requirement already satisfied: optree in /usr/local/lib/python3.10/dist-packages (from keras>=3.2.0->tensorflow>=2.15.0->mhcflurry) (0.13.1)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests<3,>=2.21.0->tensorflow>=2.15.0->mhcflurry) (3.4.0)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests<3,>=2.21.0->tensorflow>=2.15.0->mhcflurry) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests<3,>=2.21.0->tensorflow>=2.15.0->mhcflurry) (2.2.3)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests<3,>=2.21.0->tensorflow>=2.15.0->mhcflurry) (2024.12.14)\n",
            "Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.10/dist-packages (from tensorboard<2.18,>=2.17->tensorflow>=2.15.0->mhcflurry) (3.7)\n",
            "Requirement already satisfied: tensorboard-data-server<0.8.0,>=0.7.0 in /usr/local/lib/python3.10/dist-packages (from tensorboard<2.18,>=2.17->tensorflow>=2.15.0->mhcflurry) (0.7.2)\n",
            "Requirement already satisfied: werkzeug>=1.0.1 in /usr/local/lib/python3.10/dist-packages (from tensorboard<2.18,>=2.17->tensorflow>=2.15.0->mhcflurry) (3.1.3)\n",
            "Requirement already satisfied: typechecks>=0.0.2 in /usr/local/lib/python3.10/dist-packages (from serializable->mhcgnomes>=0.8.4->mhcflurry) (0.1.0)\n",
            "Requirement already satisfied: simplejson in /usr/local/lib/python3.10/dist-packages (from serializable->mhcgnomes>=0.8.4->mhcflurry) (3.19.3)\n",
            "Requirement already satisfied: MarkupSafe>=2.1.1 in /usr/local/lib/python3.10/dist-packages (from werkzeug>=1.0.1->tensorboard<2.18,>=2.17->tensorflow>=2.15.0->mhcflurry) (3.0.2)\n",
            "Requirement already satisfied: markdown-it-py>=2.2.0 in /usr/local/lib/python3.10/dist-packages (from rich->keras>=3.2.0->tensorflow>=2.15.0->mhcflurry) (3.0.0)\n",
            "Requirement already satisfied: pygments<3.0.0,>=2.13.0 in /usr/local/lib/python3.10/dist-packages (from rich->keras>=3.2.0->tensorflow>=2.15.0->mhcflurry) (2.18.0)\n",
            "Requirement already satisfied: mdurl~=0.1 in /usr/local/lib/python3.10/dist-packages (from markdown-it-py>=2.2.0->rich->keras>=3.2.0->tensorflow>=2.15.0->mhcflurry) (0.1.2)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mhcflurry-downloads fetch models_class1_pan\n",
        "!mhcflurry-downloads fetch models_class1_presentation\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xHQwNumqgoZL",
        "outputId": "afe2c909-925e-44f8-a982-e3c1e1c2e855"
      },
      "execution_count": 96,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Fetching 1/22 downloads from release 2.2.0\n",
            "DOWNLOAD NAME                             ALREADY DOWNLOADED?    WILL DOWNLOAD NOW?    URL                  \n",
            "models_class1_pan                         NO                     YES                   https://github.com/openvax/mhcflurry/releases/download/pre-2.0/models_class1_pan.selected.20200610.tar.bz2 \n",
            "models_class1_presentation                NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.0/models_class1_presentation.20200611.tar.bz2 \n",
            "models_class1_processing                  NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.0/models_class1_processing.selected.20200611.tar.bz2 \n",
            "models_class1_pan_unselected              NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.0/models_class1_pan.20200610.tar.bz2 \n",
            "models_class1_pan_variants                NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.0/models_class1_pan_variants.selected.20200610.tar.bz2 \n",
            "data_evaluation                           NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.0/data_evaluation.20200611.tar.bz2.part.aa + 2 more parts \n",
            "analysis_predictor_info                   NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.0/analysis_predictor_info.20200711.tar.bz2 \n",
            "data_predictions                          NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.0/data_predictions.20200713.tar.bz2.part.aa + 11 more parts \n",
            "data_mass_spec_annotated                  NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/1.6.0/data_mass_spec_annotated.20191226.tar.bz2 \n",
            "data_references                           NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-1.4.0/data_references.20190927.tar.bz2 \n",
            "data_iedb                                 NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.1/data_iedb.20231019.tar.bz2 \n",
            "data_systemhcatlas                        NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pan-dev1/data_systemhcatlas.20190506.tar.bz2 \n",
            "allele_sequences                          NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.1/allele_sequences.20231025.tar.bz2 \n",
            "random_peptide_predictions                NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pan-dev1/random_peptide_predictions.20190506.tar.bz2 \n",
            "data_published                            NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-1.7.0/data_published.20200501.tar.bz2 \n",
            "data_curated                              NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.1/data_curated.20231023.tar.bz2 \n",
            "models_class1                             NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pre-1.2/models_class1.20180225.tar.bz2 \n",
            "models_class1_selected_no_mass_spec       NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pre-1.2/models_class1_selected_no_mass_spec.20180225.tar.bz2 \n",
            "models_class1_unselected                  NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pre-1.2/models_class1_unselected.20180221.tar.bz2 \n",
            "models_class1_trained_with_mass_spec      NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pre-1.2.1/models_class1_trained_with_mass_spec.20180228.tar.bz2 \n",
            "models_class1_unselected_with_mass_spec   NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pre-1.2.1/models_class1_unselected_with_mass_spec.20180227.tar.bz2 \n",
            "models_class1_minimal                     NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pre-1.2/models_class1_minimal.20180226.tar.bz2 \n",
            "Downloading [part 1/1]: https://github.com/openvax/mhcflurry/releases/download/pre-2.0/models_class1_pan.selected.20200610.tar.bz2\n",
            "164MB [00:02, 72.8MB/s]               \n",
            "Downloaded to: /tmp/tmp4dd4gsmy.tar.bz2\n",
            "Extracting: 100% 27/27 [00:22<00:00,  1.18it/s]\n",
            "Extracted 27 files to: /root/.local/share/mhcflurry/4/2.2.0/models_class1_pan/\n",
            "Fetching 1/22 downloads from release 2.2.0\n",
            "DOWNLOAD NAME                             ALREADY DOWNLOADED?    WILL DOWNLOAD NOW?    URL                  \n",
            "models_class1_pan                         YES                    NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.0/models_class1_pan.selected.20200610.tar.bz2 \n",
            "models_class1_presentation                NO                     YES                   https://github.com/openvax/mhcflurry/releases/download/pre-2.0/models_class1_presentation.20200611.tar.bz2 \n",
            "models_class1_processing                  NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.0/models_class1_processing.selected.20200611.tar.bz2 \n",
            "models_class1_pan_unselected              NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.0/models_class1_pan.20200610.tar.bz2 \n",
            "models_class1_pan_variants                NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.0/models_class1_pan_variants.selected.20200610.tar.bz2 \n",
            "data_evaluation                           NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.0/data_evaluation.20200611.tar.bz2.part.aa + 2 more parts \n",
            "analysis_predictor_info                   NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.0/analysis_predictor_info.20200711.tar.bz2 \n",
            "data_predictions                          NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.0/data_predictions.20200713.tar.bz2.part.aa + 11 more parts \n",
            "data_mass_spec_annotated                  NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/1.6.0/data_mass_spec_annotated.20191226.tar.bz2 \n",
            "data_references                           NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-1.4.0/data_references.20190927.tar.bz2 \n",
            "data_iedb                                 NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.1/data_iedb.20231019.tar.bz2 \n",
            "data_systemhcatlas                        NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pan-dev1/data_systemhcatlas.20190506.tar.bz2 \n",
            "allele_sequences                          NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.1/allele_sequences.20231025.tar.bz2 \n",
            "random_peptide_predictions                NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pan-dev1/random_peptide_predictions.20190506.tar.bz2 \n",
            "data_published                            NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-1.7.0/data_published.20200501.tar.bz2 \n",
            "data_curated                              NO                     NO                    https://github.com/openvax/mhcflurry/releases/download/pre-2.1/data_curated.20231023.tar.bz2 \n",
            "models_class1                             NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pre-1.2/models_class1.20180225.tar.bz2 \n",
            "models_class1_selected_no_mass_spec       NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pre-1.2/models_class1_selected_no_mass_spec.20180225.tar.bz2 \n",
            "models_class1_unselected                  NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pre-1.2/models_class1_unselected.20180221.tar.bz2 \n",
            "models_class1_trained_with_mass_spec      NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pre-1.2.1/models_class1_trained_with_mass_spec.20180228.tar.bz2 \n",
            "models_class1_unselected_with_mass_spec   NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pre-1.2.1/models_class1_unselected_with_mass_spec.20180227.tar.bz2 \n",
            "models_class1_minimal                     NO                     NO                    http://github.com/openvax/mhcflurry/releases/download/pre-1.2/models_class1_minimal.20180226.tar.bz2 \n",
            "Downloading [part 1/1]: https://github.com/openvax/mhcflurry/releases/download/pre-2.0/models_class1_presentation.20200611.tar.bz2\n",
            "135MB [00:00, 161MB/s]               \n",
            "Downloaded to: /tmp/tmp35318wuw.tar.bz2\n",
            "Extracting: 100% 62/62 [00:17<00:00,  3.53it/s]\n",
            "Extracted 62 files to: /root/.local/share/mhcflurry/4/2.2.0/models_class1_presentation/\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pyensembl install --release 87 --species human\n",
        "!pyensembl install --release 87 --species mouse"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "n-walnYGifsH",
        "outputId": "ff7a57bb-309f-4455-abb9-4544bf884423"
      },
      "execution_count": 103,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "2025-01-03 09:50:01,073 - pyensembl.shell - INFO - Running 'install' for EnsemblRelease(release=87, species='homo_sapiens')\n",
            "2025-01-03 09:50:01,073 - pyensembl.download_cache - INFO - Fetching /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.87.gtf.gz from URL https://ftp.ensembl.org/pub/release-87/gtf/homo_sapiens/Homo_sapiens.GRCh38.87.gtf.gz\n",
            "2025-01-03 09:50:01,074 - datacache.download - INFO - Downloading https://ftp.ensembl.org/pub/release-87/gtf/homo_sapiens/Homo_sapiens.GRCh38.87.gtf.gz to /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.87.gtf.gz\n",
            "2025-01-03 09:50:56,950 - pyensembl.download_cache - INFO - Fetching /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.cdna.all.fa.gz from URL https://ftp.ensembl.org/pub/release-87/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
            "2025-01-03 09:50:56,950 - datacache.download - INFO - Downloading https://ftp.ensembl.org/pub/release-87/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz to /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
            "2025-01-03 09:52:15,800 - pyensembl.download_cache - INFO - Fetching /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.ncrna.fa.gz from URL https://ftp.ensembl.org/pub/release-87/fasta/homo_sapiens/ncrna/Homo_sapiens.GRCh38.ncrna.fa.gz\n",
            "2025-01-03 09:52:15,800 - datacache.download - INFO - Downloading https://ftp.ensembl.org/pub/release-87/fasta/homo_sapiens/ncrna/Homo_sapiens.GRCh38.ncrna.fa.gz to /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.ncrna.fa.gz\n",
            "2025-01-03 09:52:27,249 - pyensembl.download_cache - INFO - Fetching /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.pep.all.fa.gz from URL https://ftp.ensembl.org/pub/release-87/fasta/homo_sapiens/pep/Homo_sapiens.GRCh38.pep.all.fa.gz\n",
            "2025-01-03 09:52:27,249 - datacache.download - INFO - Downloading https://ftp.ensembl.org/pub/release-87/fasta/homo_sapiens/pep/Homo_sapiens.GRCh38.pep.all.fa.gz to /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.pep.all.fa.gz\n",
            "2025-01-03 09:52:44,255 - pyensembl.database - INFO - Creating database: /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.87.gtf.db\n",
            "2025-01-03 09:52:44,256 - pyensembl.database - INFO - Reading GTF from /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.87.gtf.gz\n",
            "2025-01-03 09:54:31,905 - datacache.database_helpers - INFO - Creating database /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.87.gtf.db containing: transcript, exon, gene, start_codon, CDS, stop_codon\n",
            "2025-01-03 09:54:31,905 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE transcript (protein_version TEXT NOT NULL, gene_id TEXT NOT NULL, end INT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, exon_id TEXT NOT NULL, transcript_id TEXT UNIQUE PRIMARY KEY NOT NULL, gene_version TEXT NOT NULL, feature TEXT NOT NULL, transcript_support_level TEXT NOT NULL, transcript_version TEXT NOT NULL, ccds_id TEXT NOT NULL, strand TEXT NOT NULL, start INT NOT NULL, seqname TEXT NOT NULL, gene_name TEXT NOT NULL, gene_biotype TEXT NOT NULL, exon_version TEXT NOT NULL, source TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_biotype TEXT NOT NULL)\"\n",
            "2025-01-03 09:54:32,527 - datacache.database - INFO - Inserting 198002 rows into table transcript\n",
            "2025-01-03 09:54:34,054 - datacache.database - INFO - Creating index on transcript (seqname, start, end)\n",
            "2025-01-03 09:54:34,309 - datacache.database - INFO - Creating index on transcript (gene_name)\n",
            "2025-01-03 09:54:34,559 - datacache.database - INFO - Creating index on transcript (gene_id)\n",
            "2025-01-03 09:54:34,793 - datacache.database - INFO - Creating index on transcript (transcript_name)\n",
            "2025-01-03 09:54:35,046 - datacache.database - INFO - Creating index on transcript (exon_id)\n",
            "2025-01-03 09:54:35,219 - datacache.database - INFO - Creating index on transcript (protein_id)\n",
            "2025-01-03 09:54:35,407 - datacache.database - INFO - Creating index on transcript (ccds_id)\n",
            "2025-01-03 09:54:35,621 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE exon (protein_version TEXT NOT NULL, gene_id TEXT NOT NULL, end INT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, exon_id TEXT NOT NULL, transcript_id TEXT NOT NULL, gene_version TEXT NOT NULL, feature TEXT NOT NULL, transcript_support_level TEXT NOT NULL, transcript_version TEXT NOT NULL, ccds_id TEXT NOT NULL, strand TEXT NOT NULL, start INT NOT NULL, seqname TEXT NOT NULL, gene_name TEXT NOT NULL, gene_biotype TEXT NOT NULL, exon_version TEXT NOT NULL, source TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_biotype TEXT NOT NULL)\"\n",
            "2025-01-03 09:54:39,406 - datacache.database - INFO - Inserting 1182163 rows into table exon\n",
            "2025-01-03 09:54:45,749 - datacache.database - INFO - Creating index on exon (seqname, start, end)\n",
            "2025-01-03 09:54:47,042 - datacache.database - INFO - Creating index on exon (gene_name)\n",
            "2025-01-03 09:54:48,439 - datacache.database - INFO - Creating index on exon (gene_id)\n",
            "2025-01-03 09:54:50,006 - datacache.database - INFO - Creating index on exon (transcript_id)\n",
            "2025-01-03 09:54:51,341 - datacache.database - INFO - Creating index on exon (transcript_name)\n",
            "2025-01-03 09:54:52,320 - datacache.database - INFO - Creating index on exon (exon_id)\n",
            "2025-01-03 09:54:53,510 - datacache.database - INFO - Creating index on exon (protein_id)\n",
            "2025-01-03 09:54:54,152 - datacache.database - INFO - Creating index on exon (ccds_id)\n",
            "2025-01-03 09:54:54,877 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE gene (protein_version TEXT NOT NULL, gene_id TEXT UNIQUE PRIMARY KEY NOT NULL, end INT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, exon_id TEXT NOT NULL, transcript_id TEXT NOT NULL, gene_version TEXT NOT NULL, feature TEXT NOT NULL, transcript_support_level TEXT NOT NULL, transcript_version TEXT NOT NULL, ccds_id TEXT NOT NULL, strand TEXT NOT NULL, start INT NOT NULL, seqname TEXT NOT NULL, gene_name TEXT NOT NULL, gene_biotype TEXT NOT NULL, exon_version TEXT NOT NULL, source TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_biotype TEXT NOT NULL)\"\n",
            "2025-01-03 09:54:55,027 - datacache.database - INFO - Inserting 58051 rows into table gene\n",
            "2025-01-03 09:54:55,419 - datacache.database - INFO - Creating index on gene (seqname, start, end)\n",
            "2025-01-03 09:54:55,457 - datacache.database - INFO - Creating index on gene (gene_name)\n",
            "2025-01-03 09:54:55,503 - datacache.database - INFO - Creating index on gene (transcript_id)\n",
            "2025-01-03 09:54:55,530 - datacache.database - INFO - Creating index on gene (transcript_name)\n",
            "2025-01-03 09:54:55,549 - datacache.database - INFO - Creating index on gene (exon_id)\n",
            "2025-01-03 09:54:55,567 - datacache.database - INFO - Creating index on gene (protein_id)\n",
            "2025-01-03 09:54:55,589 - datacache.database - INFO - Creating index on gene (ccds_id)\n",
            "2025-01-03 09:54:55,610 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE start_codon (protein_version TEXT NOT NULL, gene_id TEXT NOT NULL, end INT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, exon_id TEXT NOT NULL, transcript_id TEXT NOT NULL, gene_version TEXT NOT NULL, feature TEXT NOT NULL, transcript_support_level TEXT NOT NULL, transcript_version TEXT NOT NULL, ccds_id TEXT NOT NULL, strand TEXT NOT NULL, start INT NOT NULL, seqname TEXT NOT NULL, gene_name TEXT NOT NULL, gene_biotype TEXT NOT NULL, exon_version TEXT NOT NULL, source TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_biotype TEXT NOT NULL)\"\n",
            "2025-01-03 09:54:55,830 - datacache.database - INFO - Inserting 82657 rows into table start_codon\n",
            "2025-01-03 09:54:56,283 - datacache.database - INFO - Creating index on start_codon (seqname, start, end)\n",
            "2025-01-03 09:54:56,342 - datacache.database - INFO - Creating index on start_codon (gene_name)\n",
            "2025-01-03 09:54:56,397 - datacache.database - INFO - Creating index on start_codon (gene_id)\n",
            "2025-01-03 09:54:56,449 - datacache.database - INFO - Creating index on start_codon (transcript_id)\n",
            "2025-01-03 09:54:56,516 - datacache.database - INFO - Creating index on start_codon (transcript_name)\n",
            "2025-01-03 09:54:56,583 - datacache.database - INFO - Creating index on start_codon (exon_id)\n",
            "2025-01-03 09:54:56,612 - datacache.database - INFO - Creating index on start_codon (protein_id)\n",
            "2025-01-03 09:54:56,646 - datacache.database - INFO - Creating index on start_codon (ccds_id)\n",
            "2025-01-03 09:54:56,696 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE CDS (protein_version TEXT NOT NULL, gene_id TEXT NOT NULL, end INT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, exon_id TEXT NOT NULL, transcript_id TEXT NOT NULL, gene_version TEXT NOT NULL, feature TEXT NOT NULL, transcript_support_level TEXT NOT NULL, transcript_version TEXT NOT NULL, ccds_id TEXT NOT NULL, strand TEXT NOT NULL, start INT NOT NULL, seqname TEXT NOT NULL, gene_name TEXT NOT NULL, gene_biotype TEXT NOT NULL, exon_version TEXT NOT NULL, source TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_biotype TEXT NOT NULL)\"\n",
            "2025-01-03 09:54:58,462 - datacache.database - INFO - Inserting 703935 rows into table CDS\n",
            "2025-01-03 09:55:03,333 - datacache.database - INFO - Creating index on CDS (seqname, start, end)\n",
            "2025-01-03 09:55:04,403 - datacache.database - INFO - Creating index on CDS (gene_name)\n",
            "2025-01-03 09:55:04,908 - datacache.database - INFO - Creating index on CDS (gene_id)\n",
            "2025-01-03 09:55:05,404 - datacache.database - INFO - Creating index on CDS (transcript_id)\n",
            "2025-01-03 09:55:05,913 - datacache.database - INFO - Creating index on CDS (transcript_name)\n",
            "2025-01-03 09:55:06,383 - datacache.database - INFO - Creating index on CDS (exon_id)\n",
            "2025-01-03 09:55:06,721 - datacache.database - INFO - Creating index on CDS (protein_id)\n",
            "2025-01-03 09:55:07,288 - datacache.database - INFO - Creating index on CDS (ccds_id)\n",
            "2025-01-03 09:55:07,760 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE stop_codon (protein_version TEXT NOT NULL, gene_id TEXT NOT NULL, end INT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, exon_id TEXT NOT NULL, transcript_id TEXT NOT NULL, gene_version TEXT NOT NULL, feature TEXT NOT NULL, transcript_support_level TEXT NOT NULL, transcript_version TEXT NOT NULL, ccds_id TEXT NOT NULL, strand TEXT NOT NULL, start INT NOT NULL, seqname TEXT NOT NULL, gene_name TEXT NOT NULL, gene_biotype TEXT NOT NULL, exon_version TEXT NOT NULL, source TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_biotype TEXT NOT NULL)\"\n",
            "2025-01-03 09:55:08,095 - datacache.database - INFO - Inserting 74244 rows into table stop_codon\n",
            "2025-01-03 09:55:08,804 - datacache.database - INFO - Creating index on stop_codon (seqname, start, end)\n",
            "2025-01-03 09:55:08,897 - datacache.database - INFO - Creating index on stop_codon (gene_name)\n",
            "2025-01-03 09:55:08,987 - datacache.database - INFO - Creating index on stop_codon (gene_id)\n",
            "2025-01-03 09:55:09,073 - datacache.database - INFO - Creating index on stop_codon (transcript_id)\n",
            "2025-01-03 09:55:09,169 - datacache.database - INFO - Creating index on stop_codon (transcript_name)\n",
            "2025-01-03 09:55:09,270 - datacache.database - INFO - Creating index on stop_codon (exon_id)\n",
            "2025-01-03 09:55:09,328 - datacache.database - INFO - Creating index on stop_codon (protein_id)\n",
            "2025-01-03 09:55:09,388 - datacache.database - INFO - Creating index on stop_codon (ccds_id)\n",
            "2025-01-03 09:55:09,493 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE _datacache_metadata (version INT)\"\n",
            "2025-01-03 09:55:09,493 - datacache.database - INFO - Running sqlite query: \"INSERT INTO _datacache_metadata VALUES (3)\"\n",
            "2025-01-03 09:55:10,141 - pyensembl.sequence_data - INFO - Parsing sequences from FASTA file at /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.cdna.all.fa.gz\n",
            "2025-01-03 09:55:18,702 - pyensembl.sequence_data - INFO - Saving sequence dictionary to /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.cdna.all.fa.gz.pickle\n",
            "2025-01-03 09:55:19,569 - pyensembl.sequence_data - INFO - Parsing sequences from FASTA file at /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.ncrna.fa.gz\n",
            "2025-01-03 09:55:20,408 - pyensembl.sequence_data - INFO - Saving sequence dictionary to /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.ncrna.fa.gz.pickle\n",
            "2025-01-03 09:55:20,488 - pyensembl.sequence_data - INFO - Parsing sequences from FASTA file at /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.pep.all.fa.gz\n",
            "2025-01-03 09:55:23,000 - pyensembl.sequence_data - INFO - Saving sequence dictionary to /root/.cache/pyensembl/GRCh38/ensembl87/Homo_sapiens.GRCh38.pep.all.fa.gz.pickle\n",
            "2025-01-03 09:55:24,485 - pyensembl.shell - INFO - Running 'install' for EnsemblRelease(release=87, species='mus_musculus')\n",
            "2025-01-03 09:55:24,485 - pyensembl.download_cache - INFO - Fetching /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.87.gtf.gz from URL https://ftp.ensembl.org/pub/release-87/gtf/mus_musculus/Mus_musculus.GRCm38.87.gtf.gz\n",
            "2025-01-03 09:55:24,487 - datacache.download - INFO - Downloading https://ftp.ensembl.org/pub/release-87/gtf/mus_musculus/Mus_musculus.GRCm38.87.gtf.gz to /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.87.gtf.gz\n",
            "2025-01-03 09:56:00,324 - pyensembl.download_cache - INFO - Fetching /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.cdna.all.fa.gz from URL https://ftp.ensembl.org/pub/release-87/fasta/mus_musculus/cdna/Mus_musculus.GRCm38.cdna.all.fa.gz\n",
            "2025-01-03 09:56:00,324 - datacache.download - INFO - Downloading https://ftp.ensembl.org/pub/release-87/fasta/mus_musculus/cdna/Mus_musculus.GRCm38.cdna.all.fa.gz to /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.cdna.all.fa.gz\n",
            "2025-01-03 09:56:55,933 - pyensembl.download_cache - INFO - Fetching /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.ncrna.fa.gz from URL https://ftp.ensembl.org/pub/release-87/fasta/mus_musculus/ncrna/Mus_musculus.GRCm38.ncrna.fa.gz\n",
            "2025-01-03 09:56:55,933 - datacache.download - INFO - Downloading https://ftp.ensembl.org/pub/release-87/fasta/mus_musculus/ncrna/Mus_musculus.GRCm38.ncrna.fa.gz to /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.ncrna.fa.gz\n",
            "2025-01-03 09:57:02,472 - pyensembl.download_cache - INFO - Fetching /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.pep.all.fa.gz from URL https://ftp.ensembl.org/pub/release-87/fasta/mus_musculus/pep/Mus_musculus.GRCm38.pep.all.fa.gz\n",
            "2025-01-03 09:57:02,472 - datacache.download - INFO - Downloading https://ftp.ensembl.org/pub/release-87/fasta/mus_musculus/pep/Mus_musculus.GRCm38.pep.all.fa.gz to /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.pep.all.fa.gz\n",
            "2025-01-03 09:57:16,209 - pyensembl.database - INFO - Creating database: /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.87.gtf.db\n",
            "2025-01-03 09:57:16,209 - pyensembl.database - INFO - Reading GTF from /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.87.gtf.gz\n",
            "2025-01-03 09:58:24,605 - datacache.database_helpers - INFO - Creating database /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.87.gtf.db containing: CDS, exon, stop_codon, start_codon, gene, transcript\n",
            "2025-01-03 09:58:24,606 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE CDS (exon_id TEXT NOT NULL, protein_version TEXT NOT NULL, transcript_version TEXT NOT NULL, transcript_biotype TEXT NOT NULL, start INT NOT NULL, source TEXT NOT NULL, transcript_support_level TEXT NOT NULL, exon_version TEXT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, gene_id TEXT NOT NULL, strand TEXT NOT NULL, gene_version TEXT NOT NULL, ccds_id TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_id TEXT NOT NULL, seqname TEXT NOT NULL, end INT NOT NULL, feature TEXT NOT NULL, gene_name TEXT NOT NULL, gene_biotype TEXT NOT NULL)\"\n",
            "2025-01-03 09:58:27,070 - datacache.database - INFO - Inserting 484047 rows into table CDS\n",
            "2025-01-03 09:58:29,699 - datacache.database - INFO - Creating index on CDS (seqname, start, end)\n",
            "2025-01-03 09:58:30,151 - datacache.database - INFO - Creating index on CDS (gene_name)\n",
            "2025-01-03 09:58:30,478 - datacache.database - INFO - Creating index on CDS (gene_id)\n",
            "2025-01-03 09:58:30,879 - datacache.database - INFO - Creating index on CDS (transcript_id)\n",
            "2025-01-03 09:58:31,276 - datacache.database - INFO - Creating index on CDS (transcript_name)\n",
            "2025-01-03 09:58:31,615 - datacache.database - INFO - Creating index on CDS (exon_id)\n",
            "2025-01-03 09:58:31,849 - datacache.database - INFO - Creating index on CDS (protein_id)\n",
            "2025-01-03 09:58:32,264 - datacache.database - INFO - Creating index on CDS (ccds_id)\n",
            "2025-01-03 09:58:32,570 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE exon (exon_id TEXT NOT NULL, protein_version TEXT NOT NULL, transcript_version TEXT NOT NULL, transcript_biotype TEXT NOT NULL, start INT NOT NULL, source TEXT NOT NULL, transcript_support_level TEXT NOT NULL, exon_version TEXT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, gene_id TEXT NOT NULL, strand TEXT NOT NULL, gene_version TEXT NOT NULL, ccds_id TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_id TEXT NOT NULL, seqname TEXT NOT NULL, end INT NOT NULL, feature TEXT NOT NULL, gene_name TEXT NOT NULL, gene_biotype TEXT NOT NULL)\"\n",
            "2025-01-03 09:58:34,385 - datacache.database - INFO - Inserting 745096 rows into table exon\n",
            "2025-01-03 09:58:39,630 - datacache.database - INFO - Creating index on exon (seqname, start, end)\n",
            "2025-01-03 09:58:40,640 - datacache.database - INFO - Creating index on exon (gene_name)\n",
            "2025-01-03 09:58:41,223 - datacache.database - INFO - Creating index on exon (gene_id)\n",
            "2025-01-03 09:58:41,797 - datacache.database - INFO - Creating index on exon (transcript_id)\n",
            "2025-01-03 09:58:42,399 - datacache.database - INFO - Creating index on exon (transcript_name)\n",
            "2025-01-03 09:58:42,904 - datacache.database - INFO - Creating index on exon (exon_id)\n",
            "2025-01-03 09:58:43,510 - datacache.database - INFO - Creating index on exon (protein_id)\n",
            "2025-01-03 09:58:43,882 - datacache.database - INFO - Creating index on exon (ccds_id)\n",
            "2025-01-03 09:58:44,339 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE stop_codon (exon_id TEXT NOT NULL, protein_version TEXT NOT NULL, transcript_version TEXT NOT NULL, transcript_biotype TEXT NOT NULL, start INT NOT NULL, source TEXT NOT NULL, transcript_support_level TEXT NOT NULL, exon_version TEXT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, gene_id TEXT NOT NULL, strand TEXT NOT NULL, gene_version TEXT NOT NULL, ccds_id TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_id TEXT NOT NULL, seqname TEXT NOT NULL, end INT NOT NULL, feature TEXT NOT NULL, gene_name TEXT NOT NULL, gene_biotype TEXT NOT NULL)\"\n",
            "2025-01-03 09:58:44,481 - datacache.database - INFO - Inserting 50666 rows into table stop_codon\n",
            "2025-01-03 09:58:44,753 - datacache.database - INFO - Creating index on stop_codon (seqname, start, end)\n",
            "2025-01-03 09:58:44,784 - datacache.database - INFO - Creating index on stop_codon (gene_name)\n",
            "2025-01-03 09:58:44,814 - datacache.database - INFO - Creating index on stop_codon (gene_id)\n",
            "2025-01-03 09:58:44,845 - datacache.database - INFO - Creating index on stop_codon (transcript_id)\n",
            "2025-01-03 09:58:44,880 - datacache.database - INFO - Creating index on stop_codon (transcript_name)\n",
            "2025-01-03 09:58:44,913 - datacache.database - INFO - Creating index on stop_codon (exon_id)\n",
            "2025-01-03 09:58:44,930 - datacache.database - INFO - Creating index on stop_codon (protein_id)\n",
            "2025-01-03 09:58:44,949 - datacache.database - INFO - Creating index on stop_codon (ccds_id)\n",
            "2025-01-03 09:58:44,975 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE start_codon (exon_id TEXT NOT NULL, protein_version TEXT NOT NULL, transcript_version TEXT NOT NULL, transcript_biotype TEXT NOT NULL, start INT NOT NULL, source TEXT NOT NULL, transcript_support_level TEXT NOT NULL, exon_version TEXT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, gene_id TEXT NOT NULL, strand TEXT NOT NULL, gene_version TEXT NOT NULL, ccds_id TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_id TEXT NOT NULL, seqname TEXT NOT NULL, end INT NOT NULL, feature TEXT NOT NULL, gene_name TEXT NOT NULL, gene_biotype TEXT NOT NULL)\"\n",
            "2025-01-03 09:58:45,121 - datacache.database - INFO - Inserting 54619 rows into table start_codon\n",
            "2025-01-03 09:58:45,415 - datacache.database - INFO - Creating index on start_codon (seqname, start, end)\n",
            "2025-01-03 09:58:45,463 - datacache.database - INFO - Creating index on start_codon (gene_name)\n",
            "2025-01-03 09:58:45,497 - datacache.database - INFO - Creating index on start_codon (gene_id)\n",
            "2025-01-03 09:58:45,538 - datacache.database - INFO - Creating index on start_codon (transcript_id)\n",
            "2025-01-03 09:58:45,587 - datacache.database - INFO - Creating index on start_codon (transcript_name)\n",
            "2025-01-03 09:58:45,643 - datacache.database - INFO - Creating index on start_codon (exon_id)\n",
            "2025-01-03 09:58:45,664 - datacache.database - INFO - Creating index on start_codon (protein_id)\n",
            "2025-01-03 09:58:45,703 - datacache.database - INFO - Creating index on start_codon (ccds_id)\n",
            "2025-01-03 09:58:45,740 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE gene (exon_id TEXT NOT NULL, protein_version TEXT NOT NULL, transcript_version TEXT NOT NULL, transcript_biotype TEXT NOT NULL, start INT NOT NULL, source TEXT NOT NULL, transcript_support_level TEXT NOT NULL, exon_version TEXT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, gene_id TEXT UNIQUE PRIMARY KEY NOT NULL, strand TEXT NOT NULL, gene_version TEXT NOT NULL, ccds_id TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_id TEXT NOT NULL, seqname TEXT NOT NULL, end INT NOT NULL, feature TEXT NOT NULL, gene_name TEXT NOT NULL, gene_biotype TEXT NOT NULL)\"\n",
            "2025-01-03 09:58:45,854 - datacache.database - INFO - Inserting 49671 rows into table gene\n",
            "2025-01-03 09:58:46,201 - datacache.database - INFO - Creating index on gene (seqname, start, end)\n",
            "2025-01-03 09:58:46,233 - datacache.database - INFO - Creating index on gene (gene_name)\n",
            "2025-01-03 09:58:46,267 - datacache.database - INFO - Creating index on gene (transcript_id)\n",
            "2025-01-03 09:58:46,285 - datacache.database - INFO - Creating index on gene (transcript_name)\n",
            "2025-01-03 09:58:46,305 - datacache.database - INFO - Creating index on gene (exon_id)\n",
            "2025-01-03 09:58:46,322 - datacache.database - INFO - Creating index on gene (protein_id)\n",
            "2025-01-03 09:58:46,341 - datacache.database - INFO - Creating index on gene (ccds_id)\n",
            "2025-01-03 09:58:46,359 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE transcript (exon_id TEXT NOT NULL, protein_version TEXT NOT NULL, transcript_version TEXT NOT NULL, transcript_biotype TEXT NOT NULL, start INT NOT NULL, source TEXT NOT NULL, transcript_support_level TEXT NOT NULL, exon_version TEXT NOT NULL, transcript_name TEXT NOT NULL, exon_number TEXT NOT NULL, gene_id TEXT NOT NULL, strand TEXT NOT NULL, gene_version TEXT NOT NULL, ccds_id TEXT NOT NULL, protein_id TEXT NOT NULL, transcript_id TEXT UNIQUE PRIMARY KEY NOT NULL, seqname TEXT NOT NULL, end INT NOT NULL, feature TEXT NOT NULL, gene_name TEXT NOT NULL, gene_biotype TEXT NOT NULL)\"\n",
            "2025-01-03 09:58:46,683 - datacache.database - INFO - Inserting 123063 rows into table transcript\n",
            "2025-01-03 09:58:47,607 - datacache.database - INFO - Creating index on transcript (seqname, start, end)\n",
            "2025-01-03 09:58:47,695 - datacache.database - INFO - Creating index on transcript (gene_name)\n",
            "2025-01-03 09:58:47,781 - datacache.database - INFO - Creating index on transcript (gene_id)\n",
            "2025-01-03 09:58:47,874 - datacache.database - INFO - Creating index on transcript (transcript_name)\n",
            "2025-01-03 09:58:47,965 - datacache.database - INFO - Creating index on transcript (exon_id)\n",
            "2025-01-03 09:58:48,016 - datacache.database - INFO - Creating index on transcript (protein_id)\n",
            "2025-01-03 09:58:48,075 - datacache.database - INFO - Creating index on transcript (ccds_id)\n",
            "2025-01-03 09:58:48,148 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE _datacache_metadata (version INT)\"\n",
            "2025-01-03 09:58:48,148 - datacache.database - INFO - Running sqlite query: \"INSERT INTO _datacache_metadata VALUES (3)\"\n",
            "2025-01-03 09:58:48,414 - pyensembl.sequence_data - INFO - Parsing sequences from FASTA file at /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.cdna.all.fa.gz\n",
            "2025-01-03 09:58:54,123 - pyensembl.sequence_data - INFO - Saving sequence dictionary to /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.cdna.all.fa.gz.pickle\n",
            "2025-01-03 09:58:54,598 - pyensembl.sequence_data - INFO - Parsing sequences from FASTA file at /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.ncrna.fa.gz\n",
            "2025-01-03 09:58:55,044 - pyensembl.sequence_data - INFO - Saving sequence dictionary to /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.ncrna.fa.gz.pickle\n",
            "2025-01-03 09:58:55,090 - pyensembl.sequence_data - INFO - Parsing sequences from FASTA file at /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.pep.all.fa.gz\n",
            "2025-01-03 09:58:56,018 - pyensembl.sequence_data - INFO - Saving sequence dictionary to /root/.cache/pyensembl/GRCm38/ensembl87/Mus_musculus.GRCm38.pep.all.fa.gz.pickle\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pyensembl install --release 102 --species mus_musculus\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DRCCjNhdnGTG",
        "outputId": "3d67dbf6-8c10-47bd-f6e1-ade98701da13"
      },
      "execution_count": 105,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "2025-01-03 10:00:58,933 - pyensembl.shell - INFO - Running 'install' for EnsemblRelease(release=102, species='mus_musculus')\n",
            "2025-01-03 10:00:58,934 - pyensembl.database - INFO - Creating database: /root/.cache/pyensembl/GRCm38/ensembl102/Mus_musculus.GRCm38.102.gtf.db\n",
            "2025-01-03 10:00:58,934 - pyensembl.database - INFO - Reading GTF from /root/.cache/pyensembl/GRCm38/ensembl102/Mus_musculus.GRCm38.102.gtf.gz\n",
            "2025-01-03 10:02:18,763 - datacache.database_helpers - INFO - Creating database /root/.cache/pyensembl/GRCm38/ensembl102/Mus_musculus.GRCm38.102.gtf.db containing: transcript, start_codon, CDS, stop_codon, exon, gene\n",
            "2025-01-03 10:02:18,764 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE transcript (gene_biotype TEXT NOT NULL, exon_number TEXT NOT NULL, protein_version TEXT NOT NULL, strand TEXT NOT NULL, end INT NOT NULL, transcript_id TEXT UNIQUE PRIMARY KEY NOT NULL, transcript_support_level TEXT NOT NULL, start INT NOT NULL, transcript_biotype TEXT NOT NULL, transcript_name TEXT NOT NULL, gene_version TEXT NOT NULL, exon_id TEXT NOT NULL, gene_name TEXT NOT NULL, protein_id TEXT NOT NULL, source TEXT NOT NULL, feature TEXT NOT NULL, transcript_version TEXT NOT NULL, ccds_id TEXT NOT NULL, gene_id TEXT NOT NULL, exon_version TEXT NOT NULL, seqname TEXT NOT NULL)\"\n",
            "2025-01-03 10:02:19,190 - datacache.database - INFO - Inserting 142699 rows into table transcript\n",
            "2025-01-03 10:02:20,308 - datacache.database - INFO - Creating index on transcript (seqname, start, end)\n",
            "2025-01-03 10:02:20,415 - datacache.database - INFO - Creating index on transcript (gene_name)\n",
            "2025-01-03 10:02:20,516 - datacache.database - INFO - Creating index on transcript (gene_id)\n",
            "2025-01-03 10:02:20,633 - datacache.database - INFO - Creating index on transcript (transcript_name)\n",
            "2025-01-03 10:02:20,759 - datacache.database - INFO - Creating index on transcript (exon_id)\n",
            "2025-01-03 10:02:20,827 - datacache.database - INFO - Creating index on transcript (protein_id)\n",
            "2025-01-03 10:02:20,905 - datacache.database - INFO - Creating index on transcript (ccds_id)\n",
            "2025-01-03 10:02:20,994 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE start_codon (gene_biotype TEXT NOT NULL, exon_number TEXT NOT NULL, protein_version TEXT NOT NULL, strand TEXT NOT NULL, end INT NOT NULL, transcript_id TEXT NOT NULL, transcript_support_level TEXT NOT NULL, start INT NOT NULL, transcript_biotype TEXT NOT NULL, transcript_name TEXT NOT NULL, gene_version TEXT NOT NULL, exon_id TEXT NOT NULL, gene_name TEXT NOT NULL, protein_id TEXT NOT NULL, source TEXT NOT NULL, feature TEXT NOT NULL, transcript_version TEXT NOT NULL, ccds_id TEXT NOT NULL, gene_id TEXT NOT NULL, exon_version TEXT NOT NULL, seqname TEXT NOT NULL)\"\n",
            "2025-01-03 10:02:21,161 - datacache.database - INFO - Inserting 60104 rows into table start_codon\n",
            "2025-01-03 10:02:21,493 - datacache.database - INFO - Creating index on start_codon (seqname, start, end)\n",
            "2025-01-03 10:02:21,540 - datacache.database - INFO - Creating index on start_codon (gene_name)\n",
            "2025-01-03 10:02:21,577 - datacache.database - INFO - Creating index on start_codon (gene_id)\n",
            "2025-01-03 10:02:21,621 - datacache.database - INFO - Creating index on start_codon (transcript_id)\n",
            "2025-01-03 10:02:21,669 - datacache.database - INFO - Creating index on start_codon (transcript_name)\n",
            "2025-01-03 10:02:21,707 - datacache.database - INFO - Creating index on start_codon (exon_id)\n",
            "2025-01-03 10:02:21,731 - datacache.database - INFO - Creating index on start_codon (protein_id)\n",
            "2025-01-03 10:02:21,773 - datacache.database - INFO - Creating index on start_codon (ccds_id)\n",
            "2025-01-03 10:02:21,806 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE CDS (gene_biotype TEXT NOT NULL, exon_number TEXT NOT NULL, protein_version TEXT NOT NULL, strand TEXT NOT NULL, end INT NOT NULL, transcript_id TEXT NOT NULL, transcript_support_level TEXT NOT NULL, start INT NOT NULL, transcript_biotype TEXT NOT NULL, transcript_name TEXT NOT NULL, gene_version TEXT NOT NULL, exon_id TEXT NOT NULL, gene_name TEXT NOT NULL, protein_id TEXT NOT NULL, source TEXT NOT NULL, feature TEXT NOT NULL, transcript_version TEXT NOT NULL, ccds_id TEXT NOT NULL, gene_id TEXT NOT NULL, exon_version TEXT NOT NULL, seqname TEXT NOT NULL)\"\n",
            "2025-01-03 10:02:23,115 - datacache.database - INFO - Inserting 528978 rows into table CDS\n",
            "2025-01-03 10:02:26,441 - datacache.database - INFO - Creating index on CDS (seqname, start, end)\n",
            "2025-01-03 10:02:27,306 - datacache.database - INFO - Creating index on CDS (gene_name)\n",
            "2025-01-03 10:02:27,974 - datacache.database - INFO - Creating index on CDS (gene_id)\n",
            "2025-01-03 10:02:28,758 - datacache.database - INFO - Creating index on CDS (transcript_id)\n",
            "2025-01-03 10:02:29,337 - datacache.database - INFO - Creating index on CDS (transcript_name)\n",
            "2025-01-03 10:02:29,700 - datacache.database - INFO - Creating index on CDS (exon_id)\n",
            "2025-01-03 10:02:29,970 - datacache.database - INFO - Creating index on CDS (protein_id)\n",
            "2025-01-03 10:02:30,385 - datacache.database - INFO - Creating index on CDS (ccds_id)\n",
            "2025-01-03 10:02:30,726 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE stop_codon (gene_biotype TEXT NOT NULL, exon_number TEXT NOT NULL, protein_version TEXT NOT NULL, strand TEXT NOT NULL, end INT NOT NULL, transcript_id TEXT NOT NULL, transcript_support_level TEXT NOT NULL, start INT NOT NULL, transcript_biotype TEXT NOT NULL, transcript_name TEXT NOT NULL, gene_version TEXT NOT NULL, exon_id TEXT NOT NULL, gene_name TEXT NOT NULL, protein_id TEXT NOT NULL, source TEXT NOT NULL, feature TEXT NOT NULL, transcript_version TEXT NOT NULL, ccds_id TEXT NOT NULL, gene_id TEXT NOT NULL, exon_version TEXT NOT NULL, seqname TEXT NOT NULL)\"\n",
            "2025-01-03 10:02:30,878 - datacache.database - INFO - Inserting 55849 rows into table stop_codon\n",
            "2025-01-03 10:02:31,209 - datacache.database - INFO - Creating index on stop_codon (seqname, start, end)\n",
            "2025-01-03 10:02:31,245 - datacache.database - INFO - Creating index on stop_codon (gene_name)\n",
            "2025-01-03 10:02:31,279 - datacache.database - INFO - Creating index on stop_codon (gene_id)\n",
            "2025-01-03 10:02:31,321 - datacache.database - INFO - Creating index on stop_codon (transcript_id)\n",
            "2025-01-03 10:02:31,369 - datacache.database - INFO - Creating index on stop_codon (transcript_name)\n",
            "2025-01-03 10:02:31,405 - datacache.database - INFO - Creating index on stop_codon (exon_id)\n",
            "2025-01-03 10:02:31,428 - datacache.database - INFO - Creating index on stop_codon (protein_id)\n",
            "2025-01-03 10:02:31,449 - datacache.database - INFO - Creating index on stop_codon (ccds_id)\n",
            "2025-01-03 10:02:31,480 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE exon (gene_biotype TEXT NOT NULL, exon_number TEXT NOT NULL, protein_version TEXT NOT NULL, strand TEXT NOT NULL, end INT NOT NULL, transcript_id TEXT NOT NULL, transcript_support_level TEXT NOT NULL, start INT NOT NULL, transcript_biotype TEXT NOT NULL, transcript_name TEXT NOT NULL, gene_version TEXT NOT NULL, exon_id TEXT NOT NULL, gene_name TEXT NOT NULL, protein_id TEXT NOT NULL, source TEXT NOT NULL, feature TEXT NOT NULL, transcript_version TEXT NOT NULL, ccds_id TEXT NOT NULL, gene_id TEXT NOT NULL, exon_version TEXT NOT NULL, seqname TEXT NOT NULL)\"\n",
            "2025-01-03 10:02:33,522 - datacache.database - INFO - Inserting 843712 rows into table exon\n",
            "2025-01-03 10:02:38,320 - datacache.database - INFO - Creating index on exon (seqname, start, end)\n",
            "2025-01-03 10:02:39,197 - datacache.database - INFO - Creating index on exon (gene_name)\n",
            "2025-01-03 10:02:40,249 - datacache.database - INFO - Creating index on exon (gene_id)\n",
            "2025-01-03 10:02:41,511 - datacache.database - INFO - Creating index on exon (transcript_id)\n",
            "2025-01-03 10:02:42,696 - datacache.database - INFO - Creating index on exon (transcript_name)\n",
            "2025-01-03 10:02:43,351 - datacache.database - INFO - Creating index on exon (exon_id)\n",
            "2025-01-03 10:02:44,241 - datacache.database - INFO - Creating index on exon (protein_id)\n",
            "2025-01-03 10:02:44,701 - datacache.database - INFO - Creating index on exon (ccds_id)\n",
            "2025-01-03 10:02:45,259 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE gene (gene_biotype TEXT NOT NULL, exon_number TEXT NOT NULL, protein_version TEXT NOT NULL, strand TEXT NOT NULL, end INT NOT NULL, transcript_id TEXT NOT NULL, transcript_support_level TEXT NOT NULL, start INT NOT NULL, transcript_biotype TEXT NOT NULL, transcript_name TEXT NOT NULL, gene_version TEXT NOT NULL, exon_id TEXT NOT NULL, gene_name TEXT NOT NULL, protein_id TEXT NOT NULL, source TEXT NOT NULL, feature TEXT NOT NULL, transcript_version TEXT NOT NULL, ccds_id TEXT NOT NULL, gene_id TEXT UNIQUE PRIMARY KEY NOT NULL, exon_version TEXT NOT NULL, seqname TEXT NOT NULL)\"\n",
            "2025-01-03 10:02:45,388 - datacache.database - INFO - Inserting 55487 rows into table gene\n",
            "2025-01-03 10:02:45,770 - datacache.database - INFO - Creating index on gene (seqname, start, end)\n",
            "2025-01-03 10:02:45,804 - datacache.database - INFO - Creating index on gene (gene_name)\n",
            "2025-01-03 10:02:45,839 - datacache.database - INFO - Creating index on gene (transcript_id)\n",
            "2025-01-03 10:02:45,859 - datacache.database - INFO - Creating index on gene (transcript_name)\n",
            "2025-01-03 10:02:45,880 - datacache.database - INFO - Creating index on gene (exon_id)\n",
            "2025-01-03 10:02:45,905 - datacache.database - INFO - Creating index on gene (protein_id)\n",
            "2025-01-03 10:02:45,928 - datacache.database - INFO - Creating index on gene (ccds_id)\n",
            "2025-01-03 10:02:45,949 - datacache.database - INFO - Running sqlite query: \"CREATE TABLE _datacache_metadata (version INT)\"\n",
            "2025-01-03 10:02:45,949 - datacache.database - INFO - Running sqlite query: \"INSERT INTO _datacache_metadata VALUES (3)\"\n",
            "2025-01-03 10:02:46,217 - pyensembl.sequence_data - INFO - Parsing sequences from FASTA file at /root/.cache/pyensembl/GRCm38/ensembl102/Mus_musculus.GRCm38.cdna.all.fa.gz\n",
            "2025-01-03 10:02:51,458 - pyensembl.sequence_data - INFO - Saving sequence dictionary to /root/.cache/pyensembl/GRCm38/ensembl102/Mus_musculus.GRCm38.cdna.all.fa.gz.pickle\n",
            "2025-01-03 10:02:52,042 - pyensembl.sequence_data - INFO - Parsing sequences from FASTA file at /root/.cache/pyensembl/GRCm38/ensembl102/Mus_musculus.GRCm38.ncrna.fa.gz\n",
            "2025-01-03 10:02:52,695 - pyensembl.sequence_data - INFO - Saving sequence dictionary to /root/.cache/pyensembl/GRCm38/ensembl102/Mus_musculus.GRCm38.ncrna.fa.gz.pickle\n",
            "2025-01-03 10:02:52,743 - pyensembl.sequence_data - INFO - Parsing sequences from FASTA file at /root/.cache/pyensembl/GRCm38/ensembl102/Mus_musculus.GRCm38.pep.all.fa.gz\n",
            "2025-01-03 10:02:54,442 - pyensembl.sequence_data - INFO - Saving sequence dictionary to /root/.cache/pyensembl/GRCm38/ensembl102/Mus_musculus.GRCm38.pep.all.fa.gz.pickle\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!sudo apt-get install samtools\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "uoJzhBtHoOX2",
        "outputId": "ec2ba6cf-a407-4a41-84eb-aa3c663ae5ac"
      },
      "execution_count": 107,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  libhts3 libhtscodecs2\n",
            "Suggested packages:\n",
            "  cwltool\n",
            "The following NEW packages will be installed:\n",
            "  libhts3 libhtscodecs2 samtools\n",
            "0 upgraded, 3 newly installed, 0 to remove and 48 not upgraded.\n",
            "Need to get 963 kB of archives.\n",
            "After this operation, 2,270 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhtscodecs2 amd64 1.1.1-3 [53.2 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libhts3 amd64 1.13+ds-2build1 [390 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 samtools amd64 1.13-4 [520 kB]\n",
            "Fetched 963 kB in 0s (6,399 kB/s)\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 78, <> line 3.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "dpkg-preconfigure: unable to re-open stdin: \n",
            "Selecting previously unselected package libhtscodecs2:amd64.\n",
            "(Reading database ... 125903 files and directories currently installed.)\n",
            "Preparing to unpack .../libhtscodecs2_1.1.1-3_amd64.deb ...\n",
            "Unpacking libhtscodecs2:amd64 (1.1.1-3) ...\n",
            "Selecting previously unselected package libhts3:amd64.\n",
            "Preparing to unpack .../libhts3_1.13+ds-2build1_amd64.deb ...\n",
            "Unpacking libhts3:amd64 (1.13+ds-2build1) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../samtools_1.13-4_amd64.deb ...\n",
            "Unpacking samtools (1.13-4) ...\n",
            "Setting up libhtscodecs2:amd64 (1.1.1-3) ...\n",
            "Setting up libhts3:amd64 (1.13+ds-2build1) ...\n",
            "Setting up samtools (1.13-4) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.4) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libumf.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools index /content/vaxrank/tumor.bam\n"
      ],
      "metadata": {
        "id": "EcNyCBQToa8d"
      },
      "execution_count": 109,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!sudo apt-get update\n",
        "!sudo apt-get install xvfb\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "eNvvCH9Pq_EB",
        "outputId": "2cc2c584-4c4a-4c39-92e8-c65527c177d8"
      },
      "execution_count": 111,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\r0% [Working]\r            \rGet:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,626 B]\n",
            "\r0% [Connecting to archive.ubuntu.com] [Connecting to security.ubuntu.com (185.1\r                                                                               \rGet:2 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1,581 B]\n",
            "\r0% [Connecting to archive.ubuntu.com] [Connecting to security.ubuntu.com (185.1\r                                                                               \rGet:3 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]\n",
            "Get:4 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n",
            "Hit:5 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n",
            "Get:7 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  Packages [1,197 kB]\n",
            "Hit:8 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Hit:9 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Get:10 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2,632 kB]\n",
            "Hit:11 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Get:12 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [8,563 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]\n",
            "Get:14 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [2,517 kB]\n",
            "Get:15 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [1,226 kB]\n",
            "Get:16 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [3,448 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [3,633 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [2,840 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,517 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu jammy-backports/universe amd64 Packages [33.8 kB]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu jammy-backports/main amd64 Packages [81.4 kB]\n",
            "Fetched 28.1 MB in 3s (10.1 MB/s)\n",
            "Reading package lists... Done\n",
            "W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following NEW packages will be installed:\n",
            "  xvfb\n",
            "0 upgraded, 1 newly installed, 0 to remove and 54 not upgraded.\n",
            "Need to get 864 kB of archives.\n",
            "After this operation, 2,294 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 xvfb amd64 2:21.1.4-2ubuntu1.7~22.04.12 [864 kB]\n",
            "Fetched 864 kB in 1s (1,239 kB/s)\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 78, <> line 1.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "dpkg-preconfigure: unable to re-open stdin: \n",
            "Selecting previously unselected package xvfb.\n",
            "(Reading database ... 126002 files and directories currently installed.)\n",
            "Preparing to unpack .../xvfb_2%3a21.1.4-2ubuntu1.7~22.04.12_amd64.deb ...\n",
            "Unpacking xvfb (2:21.1.4-2ubuntu1.7~22.04.12) ...\n",
            "Setting up xvfb (2:21.1.4-2ubuntu1.7~22.04.12) ...\n",
            "Processing triggers for man-db (2.10.2-1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!vaxrank \\\n",
        "    --vcf /content/vaxrank/mutations.vcf \\\n",
        "    --bam /content/vaxrank/tumor.bam\\\n",
        "    --vaccine-peptide-length 25 \\\n",
        "    --mhc-predictor mhcflurry \\\n",
        "    --mhc-alleles HLA-A*01:01\\\n",
        "    --padding-around-mutation 5 \\\n",
        "    --output-ascii-report vaccine-peptides.txt \\\n",
        "    --output-pdf-report vaccine-peptides.pdf \\\n",
        "    --output-html-report vaccine-peptides.html\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "set2If6KRXQp",
        "outputId": "65b20412-f223-4ca6-8df6-dd7c41f5cf63"
      },
      "execution_count": 124,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "2025-01-03 10:35:58,357 - vaxrank.cli:407 - INFO - Namespace(vcf=['/content/vaxrank/mutations.vcf'], maf=[], variant=[], genome=None, download_reference_genome_data=False, json_variants=[], bam='/content/vaxrank/tumor.bam', min_mapping_quality=1, use_duplicate_reads=False, drop_secondary_alignments=False, num_rna_decompression_threads=1, min_variant_sequence_coverage=2, variant_sequence_assembly=True, reference_context_size=30, protein_sequence_length=20, max_reference_transcript_mismatches=2, count_mismatches_after_variant=False, min_transcript_prefix_length=10, max_protein_sequences_per_variant=1, min_alt_rna_reads=3, min_alt_rna_fragments=2, min_alt_rna_fraction=0.005, min_ratio_alt_to_other_fragments=3.0, mhc_predictor='mhcflurry', mhc_predictor_path=None, mhc_predictor_models_path=None, mhc_peptide_lengths=None, mhc_epitope_lengths=None, mhc_alleles_file=None, mhc_alleles='HLA-A*01:01', do_not_raise_on_error=False, vaccine_peptide_length=25, padding_around_mutation=5, max_vaccine_peptides_per_mutation=1, min_epitope_score=1e-10, num_epitopes_per_vaccine_peptide=None, output_patient_id='', output_csv='', output_ascii_report='vaccine-peptides.txt', output_html_report='vaccine-peptides.html', output_pdf_report='vaccine-peptides.pdf', output_json_file='', output_xlsx_report='', output_neoepitope_report='', output_reviewed_by='', output_final_review='', output_passing_variants_csv='', output_isovar_csv='', log_path='python.log', max_mutations_in_report=None, manufacturability=True, wt_epitopes=True, cosmic_vcf_filename='')\n",
            "2025-01-03 10:35:58,357 - vaxrank.cli:320 - INFO - MHC alleles: ['HLA-A*01:01']\n",
            "2025-01-03 10:35:58,365 - vaxrank.cli:323 - INFO - Variants: <VariantCollection from '/content/vaxrank/mutations.vcf' with 5 elements>\n",
            "  -- Variant(contig='11', start=101177505, ref='T', alt='C', reference_name='GRCm38')\n",
            "  -- Variant(contig='13', start=5864876, ref='', alt='G', reference_name='GRCm38')\n",
            "  -- Variant(contig='4', start=45802539, ref='G', alt='C', reference_name='GRCm38')\n",
            "  -- Variant(contig='9', start=82927102, ref='G', alt='T', reference_name='GRCm38')\n",
            "  -- Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38')\n",
            "2025-01-03 10:35:58,365 - mhctools.cli.args:179 - INFO - Building MHC binding prediction type for alleles ['HLA-A*01:01'] and epitope lengths None\n",
            "2025-01-03 10:36:04.138666: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:485] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n",
            "2025-01-03 10:36:04.157818: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:8454] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
            "2025-01-03 10:36:04.163373: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1452] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
            "2025-01-03 10:36:04.178346: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
            "To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
            "2025-01-03 10:36:05.545131: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT\n",
            "2025-01-03 10:36:10,106 - pyensembl.sequence_data:107 - INFO - Loaded sequence dictionary from /root/.cache/pyensembl/GRCm38/ensembl102/Mus_musculus.GRCm38.cdna.all.fa.gz.pickle\n",
            "2025-01-03 10:36:10,185 - pyensembl.sequence_data:107 - INFO - Loaded sequence dictionary from /root/.cache/pyensembl/GRCm38/ensembl102/Mus_musculus.GRCm38.ncrna.fa.gz.pickle\n",
            "2025-01-03 10:36:10,330 - pyensembl.sequence_data:107 - INFO - Loaded sequence dictionary from /root/.cache/pyensembl/GRCm38/ensembl102/Mus_musculus.GRCm38.pep.all.fa.gz.pickle\n",
            "2025-01-03 10:36:10,374 - vaxrank.core_logic:167 - INFO - Mutant protein fragment for Variant(contig='9', start=82927102, ref='G', alt='T', reference_name='GRCm38'): MutantProteinFragment(variant=Variant(contig='9', start=82927102, ref='G', alt='T', reference_name='GRCm38'), gene_name=Phip, amino_acids=KMKVTMVAWDRHDNTVIHAVNNMTLKVWNSYTGQL, mutant_amino_acid_start_offset=17, mutant_amino_acid_end_offset=18, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000034787', transcript_name='Phip-201', gene_id='ENSMUSG00000032253', biotype='protein_coding', contig='9', start=82866159, end=82975516, strand='-', genome='GRCm38')], n_overlapping_reads=52, n_alt_reads=15, n_ref_reads=37, n_alt_reads_supporting_protein_sequence=15)\n",
            "/usr/local/lib/python3.10/dist-packages/mhcflurry/amino_acid.py:132: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`\n",
            "  result = df.replace(letter_to_index_dict)\n",
            "1/1 [==============================] - 0s 416ms/step\n",
            "/usr/local/lib/python3.10/dist-packages/mhcflurry/amino_acid.py:132: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`\n",
            "  result = df.replace(letter_to_index_dict)\n",
            "1/1 [==============================] - 0s 39ms/step\n",
            "2025-01-03 10:36:12,199 - vaxrank.epitope_prediction:292 - INFO - 27 total peptides: 18 occur in reference, 26 failed score threshold\n",
            "2025-01-03 10:36:12,199 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='9', start=82927102, ref='G', alt='T', reference_name='GRCm38'), gene_name=Phip, amino_acids=MVAWDRHDNTVIHAVNNMTLKVWNS, mutant_amino_acid_start_offset=12, mutant_amino_acid_end_offset=13, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000034787', transcript_name='Phip-201', gene_id='ENSMUSG00000032253', biotype='protein_coding', contig='9', start=82866159, end=82975516, strand='-', genome='GRCm38')], n_overlapping_reads=52, n_alt_reads=15, n_ref_reads=37, n_alt_reads_supporting_protein_sequence=15)\n",
            "2025-01-03 10:36:12,199 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='9', start=82927102, ref='G', alt='T', reference_name='GRCm38'), gene_name=Phip, amino_acids=TMVAWDRHDNTVIHAVNNMTLKVWN, mutant_amino_acid_start_offset=13, mutant_amino_acid_end_offset=14, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000034787', transcript_name='Phip-201', gene_id='ENSMUSG00000032253', biotype='protein_coding', contig='9', start=82866159, end=82975516, strand='-', genome='GRCm38')], n_overlapping_reads=52, n_alt_reads=15, n_ref_reads=37, n_alt_reads_supporting_protein_sequence=15)\n",
            "2025-01-03 10:36:12,200 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='9', start=82927102, ref='G', alt='T', reference_name='GRCm38'), gene_name=Phip, amino_acids=VTMVAWDRHDNTVIHAVNNMTLKVW, mutant_amino_acid_start_offset=14, mutant_amino_acid_end_offset=15, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000034787', transcript_name='Phip-201', gene_id='ENSMUSG00000032253', biotype='protein_coding', contig='9', start=82866159, end=82975516, strand='-', genome='GRCm38')], n_overlapping_reads=52, n_alt_reads=15, n_ref_reads=37, n_alt_reads_supporting_protein_sequence=15)\n",
            "2025-01-03 10:36:12,200 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='9', start=82927102, ref='G', alt='T', reference_name='GRCm38'), gene_name=Phip, amino_acids=KVTMVAWDRHDNTVIHAVNNMTLKV, mutant_amino_acid_start_offset=15, mutant_amino_acid_end_offset=16, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000034787', transcript_name='Phip-201', gene_id='ENSMUSG00000032253', biotype='protein_coding', contig='9', start=82866159, end=82975516, strand='-', genome='GRCm38')], n_overlapping_reads=52, n_alt_reads=15, n_ref_reads=37, n_alt_reads_supporting_protein_sequence=15)\n",
            "2025-01-03 10:36:12,200 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='9', start=82927102, ref='G', alt='T', reference_name='GRCm38'), gene_name=Phip, amino_acids=MKVTMVAWDRHDNTVIHAVNNMTLK, mutant_amino_acid_start_offset=16, mutant_amino_acid_end_offset=17, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000034787', transcript_name='Phip-201', gene_id='ENSMUSG00000032253', biotype='protein_coding', contig='9', start=82866159, end=82975516, strand='-', genome='GRCm38')], n_overlapping_reads=52, n_alt_reads=15, n_ref_reads=37, n_alt_reads_supporting_protein_sequence=15)\n",
            "2025-01-03 10:36:12,201 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='9', start=82927102, ref='G', alt='T', reference_name='GRCm38'), gene_name=Phip, amino_acids=KMKVTMVAWDRHDNTVIHAVNNMTL, mutant_amino_acid_start_offset=17, mutant_amino_acid_end_offset=18, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000034787', transcript_name='Phip-201', gene_id='ENSMUSG00000032253', biotype='protein_coding', contig='9', start=82866159, end=82975516, strand='-', genome='GRCm38')], n_overlapping_reads=52, n_alt_reads=15, n_ref_reads=37, n_alt_reads_supporting_protein_sequence=15)\n",
            "2025-01-03 10:36:12,201 - vaxrank.core_logic:226 - INFO - Keeping 5/5 vaccine peptides for Variant(contig='9', start=82927102, ref='G', alt='T', reference_name='GRCm38')\n",
            "2025-01-03 10:36:12,203 - vaxrank.core_logic:167 - INFO - Mutant protein fragment for Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38'): MutantProteinFragment(variant=Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38'), gene_name=Wdr13, amino_acids=KAAVNKLQGHSAPVLDVIVNCDESLLASSDASGMV, mutant_amino_acid_start_offset=17, mutant_amino_acid_end_offset=18, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000033506', transcript_name='Wdr13-201', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123301, end=8132808, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000130832', transcript_name='Wdr13-205', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123324, end=8132770, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000115623', transcript_name='Wdr13-202', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8125408, end=8132820, strand='-', genome='GRCm38')], n_overlapping_reads=65, n_alt_reads=21, n_ref_reads=44, n_alt_reads_supporting_protein_sequence=21)\n",
            "/usr/local/lib/python3.10/dist-packages/mhcflurry/amino_acid.py:132: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`\n",
            "  result = df.replace(letter_to_index_dict)\n",
            "1/1 [==============================] - 0s 54ms/step\n",
            "/usr/local/lib/python3.10/dist-packages/mhcflurry/amino_acid.py:132: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`\n",
            "  result = df.replace(letter_to_index_dict)\n",
            "1/1 [==============================] - 0s 38ms/step\n",
            "2025-01-03 10:36:12,682 - vaxrank.epitope_prediction:292 - INFO - 27 total peptides: 17 occur in reference, 27 failed score threshold\n",
            "2025-01-03 10:36:12,682 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38'), gene_name=Wdr13, amino_acids=KLQGHSAPVLDVIVNCDESLLASSD, mutant_amino_acid_start_offset=12, mutant_amino_acid_end_offset=13, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000033506', transcript_name='Wdr13-201', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123301, end=8132808, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000130832', transcript_name='Wdr13-205', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123324, end=8132770, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000115623', transcript_name='Wdr13-202', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8125408, end=8132820, strand='-', genome='GRCm38')], n_overlapping_reads=65, n_alt_reads=21, n_ref_reads=44, n_alt_reads_supporting_protein_sequence=21)\n",
            "2025-01-03 10:36:12,682 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38'), gene_name=Wdr13, amino_acids=NKLQGHSAPVLDVIVNCDESLLASS, mutant_amino_acid_start_offset=13, mutant_amino_acid_end_offset=14, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000033506', transcript_name='Wdr13-201', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123301, end=8132808, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000130832', transcript_name='Wdr13-205', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123324, end=8132770, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000115623', transcript_name='Wdr13-202', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8125408, end=8132820, strand='-', genome='GRCm38')], n_overlapping_reads=65, n_alt_reads=21, n_ref_reads=44, n_alt_reads_supporting_protein_sequence=21)\n",
            "2025-01-03 10:36:12,683 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38'), gene_name=Wdr13, amino_acids=LQGHSAPVLDVIVNCDESLLASSDA, mutant_amino_acid_start_offset=11, mutant_amino_acid_end_offset=12, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000033506', transcript_name='Wdr13-201', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123301, end=8132808, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000130832', transcript_name='Wdr13-205', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123324, end=8132770, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000115623', transcript_name='Wdr13-202', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8125408, end=8132820, strand='-', genome='GRCm38')], n_overlapping_reads=65, n_alt_reads=21, n_ref_reads=44, n_alt_reads_supporting_protein_sequence=21)\n",
            "2025-01-03 10:36:12,683 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38'), gene_name=Wdr13, amino_acids=VNKLQGHSAPVLDVIVNCDESLLAS, mutant_amino_acid_start_offset=14, mutant_amino_acid_end_offset=15, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000033506', transcript_name='Wdr13-201', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123301, end=8132808, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000130832', transcript_name='Wdr13-205', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123324, end=8132770, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000115623', transcript_name='Wdr13-202', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8125408, end=8132820, strand='-', genome='GRCm38')], n_overlapping_reads=65, n_alt_reads=21, n_ref_reads=44, n_alt_reads_supporting_protein_sequence=21)\n",
            "2025-01-03 10:36:12,683 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38'), gene_name=Wdr13, amino_acids=QGHSAPVLDVIVNCDESLLASSDAS, mutant_amino_acid_start_offset=10, mutant_amino_acid_end_offset=11, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000033506', transcript_name='Wdr13-201', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123301, end=8132808, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000130832', transcript_name='Wdr13-205', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123324, end=8132770, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000115623', transcript_name='Wdr13-202', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8125408, end=8132820, strand='-', genome='GRCm38')], n_overlapping_reads=65, n_alt_reads=21, n_ref_reads=44, n_alt_reads_supporting_protein_sequence=21)\n",
            "2025-01-03 10:36:12,683 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38'), gene_name=Wdr13, amino_acids=AVNKLQGHSAPVLDVIVNCDESLLA, mutant_amino_acid_start_offset=15, mutant_amino_acid_end_offset=16, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000033506', transcript_name='Wdr13-201', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123301, end=8132808, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000130832', transcript_name='Wdr13-205', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123324, end=8132770, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000115623', transcript_name='Wdr13-202', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8125408, end=8132820, strand='-', genome='GRCm38')], n_overlapping_reads=65, n_alt_reads=21, n_ref_reads=44, n_alt_reads_supporting_protein_sequence=21)\n",
            "2025-01-03 10:36:12,683 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38'), gene_name=Wdr13, amino_acids=GHSAPVLDVIVNCDESLLASSDASG, mutant_amino_acid_start_offset=9, mutant_amino_acid_end_offset=10, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000033506', transcript_name='Wdr13-201', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123301, end=8132808, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000130832', transcript_name='Wdr13-205', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123324, end=8132770, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000115623', transcript_name='Wdr13-202', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8125408, end=8132820, strand='-', genome='GRCm38')], n_overlapping_reads=65, n_alt_reads=21, n_ref_reads=44, n_alt_reads_supporting_protein_sequence=21)\n",
            "2025-01-03 10:36:12,683 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38'), gene_name=Wdr13, amino_acids=AAVNKLQGHSAPVLDVIVNCDESLL, mutant_amino_acid_start_offset=16, mutant_amino_acid_end_offset=17, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000033506', transcript_name='Wdr13-201', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123301, end=8132808, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000130832', transcript_name='Wdr13-205', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123324, end=8132770, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000115623', transcript_name='Wdr13-202', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8125408, end=8132820, strand='-', genome='GRCm38')], n_overlapping_reads=65, n_alt_reads=21, n_ref_reads=44, n_alt_reads_supporting_protein_sequence=21)\n",
            "2025-01-03 10:36:12,683 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38'), gene_name=Wdr13, amino_acids=HSAPVLDVIVNCDESLLASSDASGM, mutant_amino_acid_start_offset=8, mutant_amino_acid_end_offset=9, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000033506', transcript_name='Wdr13-201', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123301, end=8132808, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000130832', transcript_name='Wdr13-205', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123324, end=8132770, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000115623', transcript_name='Wdr13-202', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8125408, end=8132820, strand='-', genome='GRCm38')], n_overlapping_reads=65, n_alt_reads=21, n_ref_reads=44, n_alt_reads_supporting_protein_sequence=21)\n",
            "2025-01-03 10:36:12,683 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38'), gene_name=Wdr13, amino_acids=KAAVNKLQGHSAPVLDVIVNCDESL, mutant_amino_acid_start_offset=17, mutant_amino_acid_end_offset=18, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000033506', transcript_name='Wdr13-201', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123301, end=8132808, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000130832', transcript_name='Wdr13-205', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123324, end=8132770, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000115623', transcript_name='Wdr13-202', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8125408, end=8132820, strand='-', genome='GRCm38')], n_overlapping_reads=65, n_alt_reads=21, n_ref_reads=44, n_alt_reads_supporting_protein_sequence=21)\n",
            "2025-01-03 10:36:12,684 - vaxrank.core_logic:189 - INFO - No epitope predictions for mutant protein fragment MutantProteinFragment(variant=Variant(contig='X', start=8125624, ref='C', alt='A', reference_name='GRCm38'), gene_name=Wdr13, amino_acids=SAPVLDVIVNCDESLLASSDASGMV, mutant_amino_acid_start_offset=7, mutant_amino_acid_end_offset=8, supporting_reference_transcripts=[Transcript(transcript_id='ENSMUST00000033506', transcript_name='Wdr13-201', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123301, end=8132808, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000130832', transcript_name='Wdr13-205', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8123324, end=8132770, strand='-', genome='GRCm38'), Transcript(transcript_id='ENSMUST00000115623', transcript_name='Wdr13-202', gene_id='ENSMUSG00000031166', biotype='protein_coding', contig='X', start=8125408, end=8132820, strand='-', genome='GRCm38')], n_overlapping_reads=65, n_alt_reads=21, n_ref_reads=44, n_alt_reads_supporting_protein_sequence=21)\n",
            "2025-01-03 10:36:12,684 - vaxrank.core_logic:211 - INFO - No candidate peptides for variant chrX g.8125624C>A\n",
            "2025-01-03 10:36:12,690 - vaxrank.cli:360 - INFO - About to save args: {'vcf': ['/content/vaxrank/mutations.vcf'], 'maf': [], 'variant': [], 'genome': None, 'download_reference_genome_data': False, 'json_variants': [], 'bam': '/content/vaxrank/tumor.bam', 'min_mapping_quality': 1, 'use_duplicate_reads': False, 'drop_secondary_alignments': False, 'num_rna_decompression_threads': 1, 'min_variant_sequence_coverage': 2, 'variant_sequence_assembly': True, 'reference_context_size': 30, 'protein_sequence_length': 35, 'max_reference_transcript_mismatches': 2, 'count_mismatches_after_variant': False, 'min_transcript_prefix_length': 10, 'max_protein_sequences_per_variant': 1, 'min_alt_rna_reads': 3, 'min_alt_rna_fragments': 2, 'min_alt_rna_fraction': 0.005, 'min_ratio_alt_to_other_fragments': 3.0, 'mhc_predictor': 'mhcflurry', 'mhc_predictor_path': None, 'mhc_predictor_models_path': None, 'mhc_peptide_lengths': None, 'mhc_epitope_lengths': None, 'mhc_alleles_file': None, 'mhc_alleles': 'HLA-A*01:01', 'do_not_raise_on_error': False, 'vaccine_peptide_length': 25, 'padding_around_mutation': 5, 'max_vaccine_peptides_per_mutation': 1, 'min_epitope_score': 1e-10, 'num_epitopes_per_vaccine_peptide': None, 'output_patient_id': '', 'output_csv': '', 'output_ascii_report': 'vaccine-peptides.txt', 'output_html_report': 'vaccine-peptides.html', 'output_pdf_report': 'vaccine-peptides.pdf', 'output_json_file': '', 'output_xlsx_report': '', 'output_neoepitope_report': '', 'output_reviewed_by': '', 'output_final_review': '', 'output_passing_variants_csv': '', 'output_isovar_csv': '', 'log_path': 'python.log', 'max_mutations_in_report': None, 'manufacturability': True, 'wt_epitopes': True, 'cosmic_vcf_filename': ''}\n",
            "2025-01-03 10:36:12,704 - vaxrank.report:387 - INFO - Wrote ASCII report to vaccine-peptides.txt\n",
            "2025-01-03 10:36:12,800 - vaxrank.report:394 - INFO - Wrote HTML report to vaccine-peptides.html\n",
            "2025-01-03 10:36:12,802 - vaxrank.report:413 - INFO - Running pdfkit inside xvfb wrapper\n",
            "2025-01-03 10:36:13,217 - vaxrank.report:419 - INFO - Wrote PDF report to vaccine-peptides.pdf\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from google.colab import files\n",
        "\n",
        "# لتحميل التقارير\n",
        "files.download('vaccine-peptides.txt')\n",
        "files.download('vaccine-peptides.html')\n",
        "files.download('vaccine-peptides.pdf')\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 17
        },
        "id": "F5HCj446gVOf",
        "outputId": "2f4b07ba-36e4-4302-a403-a8d910795b0a"
      },
      "execution_count": 125,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_77f5eae5-6e9b-40ec-81fc-cbba25b48ca2\", \"vaccine-peptides.txt\", 576)"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_bed03227-aee6-401f-be7d-55e4d2bd30e3\", \"vaccine-peptides.html\", 8536)"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_4258df5f-5dbe-4f88-99d8-c1a8971f2111\", \"vaccine-peptides.pdf\", 21091)"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from google.colab import auth\n",
        "auth.authenticate_user()\n"
      ],
      "metadata": {
        "id": "dcj-5J1cvkdi"
      },
      "execution_count": 1,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "9u-P0fBohnma"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}